# Glossary


# Visualizations

## Genome Browser


Upon launching the Genome Browser from the ‘Pain Study’, the user will be shown a view of the IL-2 and IL-21 genes as well as the intergenic region between them.

![](./images/BrowserOverview.png)




1. Navigation tools and track selector.  See Figure 2 for larger view.  Figure 2 element are as follows
  a.	Location/Locus entry field.  One can enter genomic coordinates in the form of chromosome:start-end (for example chr1:12345-9876), or a gene name or a SNP rs ID.
  b.	Browser zoom in and out.
  c.	Tracks: Add or hide tracks (See section below on adding/hiding tracks(
  d.	More:  Save svg image of browser, get DNA sequence or highlight regions of the browser.
2.	DNase hypersensitivity tracks.  By default, four epigenetic tracks are shown.  These are DNAse hypersensitivity tracks for Hematopoeitic stem cells (HSC), T Cells, Monocytes, and B Cells.  Additional tracks can be viewed by selecting the ‘Tracks’ button (See Adding/Removing Tracks section below)
3.	RefSeq genes.  Gene models from the RefSeq database are displayed in this tracks.
4.	-log10 of the p-value of the association of each variants with pain rate in individuals with Sickle Cell Disease.  The analysis has only been performed around the  KIAA1109/Tenr/IL2/IL21 region.   Each dot on the track represents a genomic variant (Single Nucleotide Variant (SNV) or small insertion/deletion (INDEL)).  The Y-axis for the track represents the -log10 of the p-value.  The higher the value, the more stastically significant the association between the variant and pain rate is.  Clicking on a variant will open op a window that gives further details about the variant.  (See Figure 3)
5.	-log10 of the p-value of the association of each variants with age of first vaso-occlusive crisis in individuals with Sickle Cell Disease.  See (4) above for more information on this type of track.


