



<!DOCTYPE html>
<html lang="en" class="no-js">
  <head>
    <script src="//assets.adobedtm.com/launch-EN942f79c83ff84c10a45b4856a85eaccb.min.js" async></script>
    <script async src="https://www.googletagmanager.com/gtag/js?id=UA-107924368-1"></script>
    <script>function gtag(){dataLayer.push(arguments)}window.dataLayer=window.dataLayer||[],gtag("js",new Date),gtag("config","UA-107924368-1")</script>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
      
        <link rel="canonical" href="https://www.stjude.cloud/docs/guides/tools/chipseq/">
      
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.pipeline.stopwords" content="True">
      
        <meta name="lang:search.pipeline.trimmer" content="True">
      
        <meta name="lang:search.result.none" content="No matching documents">
      
        <meta name="lang:search.result.one" content="1 matching document">
      
        <meta name="lang:search.result.other" content="# matching documents">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="../../../assets/images/favicon.ico">
      <meta name="generator" content="mkdocs-0.17.4, mkdocs-material-2.9.2">
    
    
      
        <title>ChIP-Seq Peak Calling - St. Jude Cloud Documentation</title>
      
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/application.4eccddb4.css">
      
      
    
    
      <script src="../../../assets/javascripts/modernizr.20ef595d.js"></script>

      <!-- Adobe Analytics -->
      <script src="//assets.adobedtm.com/launch-EN942f79c83ff84c10a45b4856a85eaccb.min.js" async></script>
    
    
      <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin>
      
    
    <link rel="stylesheet" href="../../../assets/fonts/material-icons.css">
    
      <link rel="stylesheet" href="../../../css/extra.css">
    
    
  </head>
  
    <body dir="ltr">
  
    <svg class="md-svg">
      <defs>
        
        
          <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448"
    viewBox="0 0 416 448" id="__github">
  <path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19-18.125
        8.5-18.125-8.5-10.75-19-3.125-20.5 3.125-20.5 10.75-19 18.125-8.5
        18.125 8.5 10.75 19 3.125 20.5zM320 304q0 10-3.125 20.5t-10.75
        19-18.125 8.5-18.125-8.5-10.75-19-3.125-20.5 3.125-20.5 10.75-19
        18.125-8.5 18.125 8.5 10.75 19 3.125 20.5zM360
        304q0-30-17.25-51t-46.75-21q-10.25 0-48.75 5.25-17.75 2.75-39.25
        2.75t-39.25-2.75q-38-5.25-48.75-5.25-29.5 0-46.75 21t-17.25 51q0 22 8
        38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0
        37.25-1.75t35-7.375 30.5-15 20.25-25.75 8-38.375zM416 260q0 51.75-15.25
        82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5-41.75
        1.125q-19.5 0-35.5-0.75t-36.875-3.125-38.125-7.5-34.25-12.875-30.25-20.25-21.5-28.75q-15.5-30.75-15.5-82.75
        0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25
        30.875q36.75-8.75 77.25-8.75 37 0 70 8 26.25-20.5
        46.75-30.25t47.25-9.75q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34
        99.5z" />
</svg>
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" data-md-component="overlay" for="__drawer"></label>
    
      <a href="#overview" tabindex="1" class="md-skip">
        Skip to content
      </a>
    
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href="https://www.stjude.cloud/docs/" title="St. Jude Cloud Documentation" class="md-header-nav__button md-logo">
          
            <img src="../../../images/stjude-logo-child-white.png" width="40" height="28">
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="__drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <div class="md-flex__ellipsis md-header-nav__title" data-md-component="title">
          
            
              <span class="md-header-nav__topic">
                St. Jude Cloud Documentation
              </span>
              <span class="md-header-nav__topic">
                ChIP-Seq Peak Calling
              </span>
            
          
        </div>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          
            <label class="md-icon md-icon--search md-header-nav__button" for="__search"></label>
            
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query" data-md-state="active">
      <label class="md-icon md-search__icon" for="__search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">
        &#xE5CD;
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
          
        
      </div>
      
        <div class="md-flex__cell md-flex__cell--shrink">
          <div class="md-header-nav__source">
            


  


  <a href="https://github.com/stjudecloud/docs/" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#__github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      stjudecloud/docs
    </div>
  </a>

          </div>
        </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
        
      
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="__drawer">
    <span class="md-nav__button md-logo">
      
        <img src="../../../images/stjude-logo-child-white.png" width="48" height="48">
      
    </span>
    St. Jude Cloud Documentation
  </label>
  
    <div class="md-nav__source">
      


  


  <a href="https://github.com/stjudecloud/docs/" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#__github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      stjudecloud/docs
    </div>
  </a>

    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="../../.." title="Welcome to St. Jude Cloud" class="md-nav__link">
      Welcome to St. Jude Cloud
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../create-an-account/" title="Create An Account" class="md-nav__link">
      Create An Account
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-3" type="checkbox" id="nav-3">
    
    <label class="md-nav__link" for="nav-3">
      Data Guides
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-3">
        Data Guides
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../data/about-our-data/" title="About Our Data" class="md-nav__link">
      About Our Data
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../data/data-request/" title="Making a Data Request" class="md-nav__link">
      Making a Data Request
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../data/working-with-our-data/" title="Working with Our Data" class="md-nav__link">
      Working with Our Data
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../data/creating-a-cloud-app/" title="Creating a Cloud App" class="md-nav__link">
      Creating a Cloud App
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../data/data-transfer-app/" title="Data Transfer Application" class="md-nav__link">
      Data Transfer Application
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../data/command-line/" title="Command Line Interaction" class="md-nav__link">
      Command Line Interaction
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-4" type="checkbox" id="nav-4" checked>
    
    <label class="md-nav__link" for="nav-4">
      Tool Guides
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-4">
        Tool Guides
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../pecan-pie/" title="Pecan PIE" class="md-nav__link">
      Pecan PIE
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../neoepitope/" title="NeoepitopePred" class="md-nav__link">
      NeoepitopePred
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="__toc">
    
    
      <label class="md-nav__link md-nav__link--active" for="__toc">
        ChIP-Seq Peak Calling
      </label>
    
    <a href="./" title="ChIP-Seq Peak Calling" class="md-nav__link md-nav__link--active">
      ChIP-Seq Peak Calling
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#overview" title="Overview" class="md-nav__link">
    Overview
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#getting-started" title="Getting started" class="md-nav__link">
    Getting started
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#uploading-data" title="Uploading data" class="md-nav__link">
    Uploading data
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#running-the-tool" title="Running the tool" class="md-nav__link">
    Running the tool
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#broad-vs-narrow-peak-calling" title="Broad vs. narrow peak calling" class="md-nav__link">
    Broad vs. narrow peak calling
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#selecting-parameters" title="Selecting parameters" class="md-nav__link">
    Selecting parameters
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hooking-up-inputs" title="Hooking up inputs" class="md-nav__link">
    Hooking up inputs
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#starting-the-workflow" title="Starting the workflow" class="md-nav__link">
    Starting the workflow
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#monitoring-run-progress" title="Monitoring run progress" class="md-nav__link">
    Monitoring run progress
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#interactive-visualizations" title="Interactive visualizations" class="md-nav__link">
    Interactive visualizations
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#finding-the-raw-results-files" title="Finding the raw results files" class="md-nav__link">
    Finding the raw results files
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#interpreting-results" title="Interpreting results" class="md-nav__link">
    Interpreting results
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#frequently-asked-questions" title="Frequently asked questions" class="md-nav__link">
    Frequently asked questions
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
    
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../rapid-rnaseq/" title="Rapid RNA-Seq Fusion Detection" class="md-nav__link">
      Rapid RNA-Seq Fusion Detection
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../warden/" title="WARDEN Differential Expression Analysis" class="md-nav__link">
      WARDEN Differential Expression Analysis
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../mutational-signatures/" title="Mutational Signatures" class="md-nav__link">
      Mutational Signatures
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../cis-x/" title="" class="md-nav__link">
      
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-5" type="checkbox" id="nav-5">
    
    <label class="md-nav__link" for="nav-5">
      Visualization Guides
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-5">
        Visualization Guides
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../portals/pecan/" title="PeCan Data Portal" class="md-nav__link">
      PeCan Data Portal
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../portals/genome-paint/" title="Genome Paint" class="md-nav__link">
      Genome Paint
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../portals/sickle-cell/" title="Sickle Cell Genomics Portal" class="md-nav__link">
      Sickle Cell Genomics Portal
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-6" type="checkbox" id="nav-6">
    
    <label class="md-nav__link" for="nav-6">
      Forms
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-6">
        Forms
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../forms/how-to-fill-out-DAA/" title="Filling out a Data Access Agreement" class="md-nav__link">
      Filling out a Data Access Agreement
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../forms/how-to-fill-out-Extension/" title="Filling out an Extension Addendum" class="md-nav__link">
      Filling out an Extension Addendum
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../citing-stjude-cloud/" title="Citing St. Jude Cloud" class="md-nav__link">
      Citing St. Jude Cloud
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../faq/" title="Frequently Asked Questions" class="md-nav__link">
      Frequently Asked Questions
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../glossary/" title="Glossary" class="md-nav__link">
      Glossary
    </a>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#overview" title="Overview" class="md-nav__link">
    Overview
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#getting-started" title="Getting started" class="md-nav__link">
    Getting started
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#uploading-data" title="Uploading data" class="md-nav__link">
    Uploading data
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#running-the-tool" title="Running the tool" class="md-nav__link">
    Running the tool
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#broad-vs-narrow-peak-calling" title="Broad vs. narrow peak calling" class="md-nav__link">
    Broad vs. narrow peak calling
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#selecting-parameters" title="Selecting parameters" class="md-nav__link">
    Selecting parameters
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hooking-up-inputs" title="Hooking up inputs" class="md-nav__link">
    Hooking up inputs
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#starting-the-workflow" title="Starting the workflow" class="md-nav__link">
    Starting the workflow
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#monitoring-run-progress" title="Monitoring run progress" class="md-nav__link">
    Monitoring run progress
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#interactive-visualizations" title="Interactive visualizations" class="md-nav__link">
    Interactive visualizations
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#finding-the-raw-results-files" title="Finding the raw results files" class="md-nav__link">
    Finding the raw results files
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#interpreting-results" title="Interpreting results" class="md-nav__link">
    Interpreting results
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#frequently-asked-questions" title="Frequently asked questions" class="md-nav__link">
    Frequently asked questions
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/stjudecloud/docs/edit/master/docs/guides/tools/chipseq.md" title="Edit this page" class="md-icon md-content__icon">&#xE3C9;</a>
                
                
                  <h1>ChIP-Seq Peak Calling</h1>
                
                <table>
<thead>
<tr>
<th></th>
<th></th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>Authors</strong></td>
<td>Xing Tang, Yong Cheng</td>
</tr>
<tr>
<td><strong>Publication</strong></td>
<td>N/A (not published)</td>
</tr>
<tr>
<td><strong>Technical Support</strong></td>
<td><a href="https://stjude.cloud/contact">Contact Us</a></td>
</tr>
</tbody>
</table>
<p>The ChIP-Seq Peak Calling workflow follows ENCODE best practices to call 
broad or narrow peaks on Illumina-generated ChIP-Seq data. 
Here, a Gzipped FastQ file from an Immunoprecipitation (IP) experiment 
is considered the "case sample file" and a Gzipped FastQ file from a control 
experiment is considered the "control sample file". The pipeline can run on
matched case/control samples (recommended for better results) or just a 
case sample.</p>
<h2 id="overview">Overview</h2>
<p><strong>Inputs</strong></p>
<table>
<thead>
<tr>
<th>Name</th>
<th>Type</th>
<th>Description</th>
<th>Example</th>
</tr>
</thead>
<tbody>
<tr>
<td>FastQ files (<em>required</em> if using FastQ inputs)</td>
<td>Input file</td>
<td>Gzipped FastQ files generated by experiment.</td>
<td>Sample_R1.fastq.gz and Sample_R2.fastq.gz</td>
</tr>
</tbody>
</table>
<p><strong>Outputs</strong></p>
<table>
<thead>
<tr>
<th>Name</th>
<th>Format</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>BED file</td>
<td>.bed</td>
<td>Peak calls</td>
</tr>
<tr>
<td>Binary file</td>
<td>.bb</td>
<td>Binary format for BED file</td>
</tr>
<tr>
<td>BigWig file</td>
<td>.bw</td>
<td>Shows read coverage</td>
</tr>
<tr>
<td>Metrics file</td>
<td>.txt</td>
<td>Shows mapping and duplication rate</td>
</tr>
<tr>
<td>Cross correlation plot</td>
<td>.pdf</td>
<td>Quality plot showing if the forward and reverse reads tend to be centered around binding sites.</td>
</tr>
</tbody>
</table>
<p><strong>Process</strong></p>
<ol>
<li>The reads of the FastQ file(s) are aligned to the specified reference genome. </li>
<li>The aligned reads are then post-processed based on best-practice QC techniques
(removing multiple mapped reads, removing duplicated reads, etc). </li>
<li>Peaks are called by SICER (broad peak analysis) or MACS2 (narrow peak
analysis). <ul>
<li>Qualified peaks will be output as BED (.bed) and big BED (.bb)
files. </li>
<li>The coverage information will be output as a bigWig (.bw)
file. </li>
<li>A cross correlation plot and general metrics file are generated to help check
the quality of experiment.</li>
</ul>
</li>
</ol>
<h2 id="getting-started">Getting started</h2>
<p>To get started, you need to navigate to the <a href="https://platform.stjude.cloud/tools/chip-seq">ChIP-Seq tool page</a>. You'll need to click
the "Start" button in the left hand pane. This creates a cloud workspace
in DNAnexus with the same name as the tool. After this, you will be able 
to upload your input files to that workspace.</p>
<p><img alt="" src="../../../images/guides/tools/chipseq/click-start.gif" /></p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If you can't see the "Start" button, one of these two scenarios is likely the case:</p>
<ul>
<li>You see three buttons on the left sidebar instead of one. In this case,
  you've already clicked the "Start" button previously, and a cloud workspace has
  already been created for you. In this case, you're good! You can move
  on to the next section.</li>
<li>If you cannot see <em>any</em> buttons on the left side, you probably have not
  logged in yet. If you see a sentence that says "Log in to launch this 
  tool", simply login and try again.</li>
</ul>
<p>If neither of these are the case and you still can't click "Start",
<a href="https://stjude.cloud/contact">contact us</a>.</p>
</div>
<h2 id="uploading-data">Uploading data</h2>
<p>The ChIP-Seq Peak Caller takes Gzipped FastQ files generated from an
IP experiment as input. You can upload your input FastQ files by
using the <a href="../../data/data-transfer-app/">data transfer application</a>
or by uploading them through <a href="../../data/command-line/">the command line</a>.
Both of the guides linked here will contain more details on how to upload
data using that method, so we defer to those guides here.</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>If you plan to upload data through the St. Jude Cloud Data Transfer application
(recommended), you can click the "Upload Data" button in the left panel. If you
have not already downloaded the app, do so by clicking "Download app". Once you
have the app, you can click "Open app" to open the app with the tool's cloud 
workspace already opened and ready to drag-and-drop files into it!</p>
<p>For more information, check out the <a href="../../data/data-transfer-app/">data transfer application</a> guide.</p>
</div>
<h2 id="running-the-tool">Running the tool</h2>
<p>Once you've uploaded data to your cloud workspace, 
click "Launch Tool" on the <a href="https://platform.stjude.cloud/tools/chip-seq">tool's landing page</a>. 
A dropdown will present the different presets for running the ChIP-Seq workflow.
You'll need to decide <strong>(1)</strong> whether you'd like to run broad/narrow peak
calling and <strong>(2)</strong> whether you have a case sample and a control sample (preferred)
or just a case sample. This will determine which preset you should
click in this dropdown. There are various other parameters that you can 
set, but they are covered in further sections of this guide.</p>
<p><img alt="" src="../../../images/guides/tools/chipseq/launch-tool.gif" /></p>
<h3 id="broad-vs-narrow-peak-calling">Broad vs. narrow peak calling</h3>
<p>Choosing between broad and narrow peak calling depends on the experiment
design. The following are good rules of thumb for choosing between the
two configurations. If you are not sure which configuration to use,
please consult with an expert at your institution or <a href="https://stjude.cloud/contact">contact us</a>.</p>
<p><strong>Narrow Peak Calling</strong></p>
<p>If your target protein is a transcription factor, you should probably
choose narrow peak calling. You can also try the narrow peak calling
workflows for the following histone marks:</p>
<ul>
<li>H3K4me3</li>
<li>H3K4me2</li>
<li>H3K9-14ac</li>
<li>H3K27ac</li>
<li>H2A.Z</li>
</ul>
<p><strong>Broad Peak Calling</strong></p>
<p>You should try the broad peak calling workflows for the following
histone marks:</p>
<ul>
<li>H3K36me3</li>
<li>H3K79me2</li>
<li>H3K27me3</li>
<li>H3K9me3</li>
<li>H3K9me1</li>
</ul>
<p><strong>Special Cases</strong></p>
<p>In some scenarios, H3K4me1, H3K9me2 and H3K9me3 might behave between
narrow and broad shape, you might need to look into each peak region and
consult experts.</p>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>If your fragment size is less than 50 base pairs, please refer to the
<a href="#frequently-asked-questions">frequently asked questions</a>.</p>
</div>
<h3 id="selecting-parameters">Selecting parameters</h3>
<p>There are a number of other parameters that can be customized. To 
see the options available, click the gear cog next to the 
"Parameter Wrapper" substep.</p>
<p><img alt="" src="../../../images/guides/tools/chipseq/change-parameters.gif" /></p>
<p>The following are the parameters that can be set, a short
description of each parameter, and an example value. If you
have questions, please <a href="https://stjude.cloud/contact">contact us</a>.</p>
<table>
<thead>
<tr>
<th>Parameter Name</th>
<th>Description</th>
<th>Example</th>
</tr>
</thead>
<tbody>
<tr>
<td>Output prefix (<em>required</em>)</td>
<td>A name used a prefix for all outputs in the run</td>
<td>SAMPLE1</td>
</tr>
<tr>
<td>Reference genome (<em>required</em>)</td>
<td>Supported reference genome from one of hg19, GRCh38, mm9, mm10, dm3</td>
<td>GRCh38</td>
</tr>
<tr>
<td>Output bigWig</td>
<td>Whether or not to include a bigwig file in the output</td>
<td>True</td>
</tr>
<tr>
<td>Remove blacklist peaks</td>
<td>Whether or not to remove known problem areas</td>
<td>True</td>
</tr>
<tr>
<td>Fragment length</td>
<td>Hardcoded fragment length of your reads. 'NA' for auto-detect.</td>
<td>NA</td>
</tr>
</tbody>
</table>
<div class="admonition caution">
<p class="admonition-title">Caution</p>
<p>Please be aware of the following stumbling points when setting parameters:</p>
<ul>
<li>Do not use spaces anywhere in your input file names, your output
  prefix, or any of the other parameters. This is generally bad
  practice and doesn't play well with the pipeline (consider using
  "_" instead).</li>
<li>Do not change the output directory when you run the pipeline. At
  the top of parameter input page, there is a text box that allows
  you to change the output folder. <strong>Please ignore that setting</strong>. You
  only need to specify an output prefix as described above. All of
  the results will be put under <code class="codehilite"><span class="o">/</span><span class="n">Results</span><span class="o">/[</span><span class="n">OUTPUT_PREFIX</span><span class="o">]</span><span class="w"></span></code>.</li>
</ul>
</div>
<h3 id="hooking-up-inputs">Hooking up inputs</h3>
<p>Next, you'll need to hook up the FastQ files you uploaded in 
<a href="#uploading-data">the upload data section</a>. You can do this by 
clicking on the <code class="codehilite"><span class="err">ChIP Reads</span></code> and <code class="codehilite"><span class="err">Control Reads</span></code> slots and
selecting the respective files. If you are not doing a case/control
run, you only need to hook up the case sample.</p>
<p><img alt="" src="../../../images/guides/tools/chipseq/hookup-inputs.gif" /></p>
<h3 id="starting-the-workflow">Starting the workflow</h3>
<p>Once your input files are hooked up, you should be able to start the workflow
by clicking the "Run as Analysis..." button in the top right hand corner
of the workflow dialog.</p>
<p><img alt="" src="../../../images/guides/tools/chipseq/run-analysis.gif" /></p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>If you cannot click this button, please ensure that:</p>
<ul>
<li>all of the inputs are correctly hooked up (see <a href="#hooking-up-inputs">hooking up inputs</a>), and </li>
<li>all of the required parameters are set (see <a href="#setting-parameters">setting parameters</a>).</li>
</ul>
<p>If you're still have trouble, please <a href="https://stjude.cloud/contact">contact us</a> and include
a screenshot of the workflow screen above.</p>
</div>
<h2 id="monitoring-run-progress">Monitoring run progress</h2>
<p>Once you have started one or more ChIP-Seq runs, you can safely close your
browser and come back later to check the status of the jobs. To do this,
navigate to the <a href="https://platform.stjude.cloud/tools/chip-seq">tool's landing page</a>. 
Next, click "View Results" then select the "View Running Jobs" option. 
You will be redirected to the job monitoring page. Each job you kicked off
gets one row in this table.</p>
<p><img alt="" src="../../../images/guides/tools/chipseq/monitoring-jobs.gif" /> </p>
<p>You can click the "+" on any of the runs to check 
the status of individual steps of the ChIP-Seq pipeline.
Other information, such as time, cost of individual steps in the pipeline, 
and the job logs can be accessed by clicking around the sub-items.</p>
<p><img alt="" src="../../../images/guides/tools/chipseq/job-detailed-view.gif" /> </p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>Power users can refer to the <a href="https://documentation.dnanexus.com/user/running-apps-and-workflows/monitoring-executions">DNAnexus Monitoring Executions Documentation</a> for advanced capabilities for monitoring jobs.</p>
</div>
<h3 id="interactive-visualizations">Interactive visualizations</h3>
<p>Today, the ChIP-Seq pipeline does not produce an interactive visualization. We are
working on adding this! In the meantime, you can view the cross-correlation plot(s)
as outlined in the sections below.</p>
<h3 id="finding-the-raw-results-files">Finding the raw results files</h3>
<p>Navigate to the <a href="https://platform.stjude.cloud/tools/chip-seq">tool's landing page</a>. 
In the left hand pane, click "View Results" then "View Results Files". You will
be taken to the filesystem view of your cloud workspace. This is similar to the
filesystem on your computer, and you can do many common operations such as deleting,
renaming, and moving files. To access ChIP-Seq results, you should click on the 
<code class="codehilite"><span class="err">Results</span></code> folder, then select the output folder name you gave in the <a href="#selecting-parameters">selecting parameters</a> part of the guide.</p>
<p><img alt="" src="../../../images/guides/tools/chipseq/view-results-files.gif" /></p>
<h3 id="interpreting-results">Interpreting results</h3>
<p>For the ChIP-Seq pipeline, every pipeline run outputs a <code class="codehilite"><span class="err">README.doc</span></code> file
which contains the latest information on which results are included.
You can refer to that file for the most up to date information on raw outputs.</p>
<p><img alt="" src="../../../images/guides/tools/chipseq/click-readme.gif" /></p>
<h2 id="frequently-asked-questions">Frequently asked questions</h2>
<p>If you have any questions not covered here, feel free to 
<a href="https://hospital.stjude.org/apps/forms/fb/st-jude-cloud-contact/">contact us</a>.</p>
<p><strong>Q: Should I choose narrow peak calling pipeline or broad peak calling pipeline?</strong></p>
<p>A. We built two workflows: one for narrow peak calling and another broad
peak calling. If your target protein is a transcription factor, please
use narrow peak calling workflow. For histone marks H3K4me3, H3K4me2,
H3K9-14ac, H3K27ac and H2A.Z, you could try narrow peak calling
workflow. For histone marks H3K36me3, H3K79me2, H3K27me3, H3K9me3 and
H3K9me1, you could try broad peak calling workflow. In some scenario,
H3K4me1, H3K9me2 and H3K9me3 might behave between narrow and broad
shape, you might need to look into each peak region and consult
experts.</p>
<p><strong>Q. What to do if your fragment size is less than 50 base pairs?</strong></p>
<p>A. We estimate fragment size from the data based on the cross correlation
plot. Usually the fragment size is above 50bp. If the estimated
fragment size lower than 50bp, the workflow will stop at the peak
calling stage (MACS2/SICER) after BWA mapping finishes. You can rerun
the analysis with a specified fragment length.</p>
                
                  
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../neoepitope/" title="NeoepitopePred" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                NeoepitopePred
              </span>
            </div>
          </a>
        
        
          <a href="../rapid-rnaseq/" title="Rapid RNA-Seq Fusion Detection" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                Rapid RNA-Seq Fusion Detection
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            Copyright &copy; 2017 - 2019 St. Jude Children's Research Hospital
          </div>
        
        
        built from <a href="https://github.com/stjudecloud/docs/tree/d82993c">d82993c</a> using
        
        <a href="https://www.mkdocs.org">MkDocs</a>
        and a modified version of
        <a href="https://squidfunk.github.io/mkdocs-material/">
          Material for MkDocs</a>.
      </div>
      
        
  <div class="md-footer-social">
    <link rel="stylesheet" href="../../../assets/fonts/font-awesome.css">
    
      <a href="https://www.facebook.com/stjuderesearch/" class="md-footer-social__link fa fa-facebook"></a>
    
      <a href="https://twitter.com/stjuderesearch?lang=en" class="md-footer-social__link fa fa-twitter"></a>
    
      <a href="https://www.linkedin.com/company/st-jude-childrens-research-hospital" class="md-footer-social__link fa fa-linkedin"></a>
    
      <a href="https://github.com/stjude" class="md-footer-social__link fa fa-github"></a>
    
  </div>

      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../../assets/javascripts/application.ec5be3af.js"></script>
      
      <script>app.initialize({version:"0.17.4",url:{base:"../../.."}})</script>
      
    
    
      
    
  </body>
</html>