



<!doctype html>
<html lang="en" class="no-js">
  <head>
    <script src="//assets.adobedtm.com/launch-EN942f79c83ff84c10a45b4856a85eaccb.min.js" async></script>
    <script>!function(e,a,t,n,g,c,o){e.GoogleAnalyticsObject=g,e.ga=e.ga||function(){(e.ga.q=e.ga.q||[]).push(arguments)},e.ga.l=1*new Date,c=a.createElement(t),o=a.getElementsByTagName(t)[0],c.async=1,c.src="https://www.google-analytics.com/analytics.js",o.parentNode.insertBefore(c,o)}(window,document,"script",0,"ga"),ga("create","UA-107924368-1","auto"),ga("send","pageview")</script>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
      
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.pipeline.stopwords" content="True">
      
        <meta name="lang:search.pipeline.trimmer" content="True">
      
        <meta name="lang:search.result.none" content="No matching documents">
      
        <meta name="lang:search.result.one" content="1 matching document">
      
        <meta name="lang:search.result.other" content="# matching documents">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="../../../../images/favicon.ico">
      <meta name="generator" content="mkdocs-1.1, mkdocs-material-4.6.3">
    
    
      
        <title>cis-X - St. Jude Cloud Documentation</title>
      
    
    
      <link rel="stylesheet" href="../../../../assets/stylesheets/application.7d2ef1ff.css">
      
      
    
    
      <script src="../../../../assets/javascripts/modernizr.eca31fed.js"></script>
    
    
      
    
    <link rel="stylesheet" href="../../../../assets/fonts/material-icons.css">
    
    
      <link rel="stylesheet" href="../../../../css/extra.css">
    
    
      
    
    
  </head>
  
    <body dir="ltr">
  
    <svg class="md-svg">
      <defs>
        
        
          <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448" viewBox="0 0 416 448" id="__github"><path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19T128 352t-18.125-8.5-10.75-19T96 304t3.125-20.5 10.75-19T128 256t18.125 8.5 10.75 19T160 304zm160 0q0 10-3.125 20.5t-10.75 19T288 352t-18.125-8.5-10.75-19T256 304t3.125-20.5 10.75-19T288 256t18.125 8.5 10.75 19T320 304zm40 0q0-30-17.25-51T296 232q-10.25 0-48.75 5.25Q229.5 240 208 240t-39.25-2.75Q130.75 232 120 232q-29.5 0-46.75 21T56 304q0 22 8 38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0 37.25-1.75t35-7.375 30.5-15 20.25-25.75T360 304zm56-44q0 51.75-15.25 82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5T212 416q-19.5 0-35.5-.75t-36.875-3.125-38.125-7.5-34.25-12.875T37 371.5t-21.5-28.75Q0 312 0 260q0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25 30.875Q171.5 96 212 96q37 0 70 8 26.25-20.5 46.75-30.25T376 64q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34 99.5z"/></svg>
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" data-md-component="overlay" for="__drawer"></label>
    
      <a href="#overview" tabindex="0" class="md-skip">
        Skip to content
      </a>
    
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href="../../../.." title="St. Jude Cloud Documentation" aria-label="St. Jude Cloud Documentation" class="md-header-nav__button md-logo">
          
            <img alt="logo" src="../../../../images/stjude-logo-child-white.png" width="40" height="28">
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="__drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <div class="md-flex__ellipsis md-header-nav__title" data-md-component="title">
          
            <span class="md-header-nav__topic">
              St. Jude Cloud Documentation
            </span>
            <span class="md-header-nav__topic">
              
                cis-X
              
            </span>
          
        </div>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <label class="md-icon md-icon--search md-header-nav__button" for="__search"></label>
          
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" aria-label="search" name="query" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query" data-md-state="active">
      <label class="md-icon md-search__icon" for="__search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">
        &#xE5CD;
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
        
      </div>
      
        <div class="md-flex__cell md-flex__cell--shrink">
          <div class="md-header-nav__source">
            


  

<a href="https://github.com/stjudecloud/docs/" title="Go to repository" class="md-source" data-md-source="github">
  
    <div class="md-source__icon">
      <svg viewBox="0 0 24 24" width="24" height="24">
        <use xlink:href="#__github" width="24" height="24"></use>
      </svg>
    </div>
  
  <div class="md-source__repository">
    stjudecloud/docs
  </div>
</a>
          </div>
        </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
        
      
      
        

  

<nav class="md-tabs md-tabs--active" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  <li class="md-tabs__item">
    
      <a href="../../../.." class="md-tabs__link">
        Overview
      </a>
    
  </li>

      
        
      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../" class="md-tabs__link md-tabs__link--active">
          Genomics Platform
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../../pecan/" class="md-tabs__link">
          PeCan
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../../visualization-community/" class="md-tabs__link">
          Visualization Community
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../../cstn/" class="md-tabs__link">
          Portals
        </a>
      
    </li>
  

      
        
  
  
    
    
  
  
    <li class="md-tabs__item">
      
        <a href="../../../proteinpaint/" class="md-tabs__link">
          ProteinPaint
        </a>
      
    </li>
  

  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../../studies/" class="md-tabs__link">
          Studies
        </a>
      
    </li>
  

      
        
      
        
      
    </ul>
  </div>
</nav>
      
      <main class="md-main" role="main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="__drawer">
    <a href="../../../.." title="St. Jude Cloud Documentation" class="md-nav__button md-logo">
      
        <img alt="logo" src="../../../../images/stjude-logo-child-white.png" width="48" height="48">
      
    </a>
    St. Jude Cloud Documentation
  </label>
  
    <div class="md-nav__source">
      


  

<a href="https://github.com/stjudecloud/docs/" title="Go to repository" class="md-source" data-md-source="github">
  
    <div class="md-source__icon">
      <svg viewBox="0 0 24 24" width="24" height="24">
        <use xlink:href="#__github" width="24" height="24"></use>
      </svg>
    </div>
  
  <div class="md-source__repository">
    stjudecloud/docs
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="../../../.." title="Overview" class="md-nav__link">
      Overview
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../../ecosystem/" title="About our Ecosystem" class="md-nav__link">
      About our Ecosystem
    </a>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-3" type="checkbox" id="nav-3" checked>
    
    <label class="md-nav__link" for="nav-3">
      Genomics Platform
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-3">
        Genomics Platform
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../" title="Getting Started" class="md-nav__link">
      Getting Started
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../accounts-and-billing/" title="Accounts and Billing" class="md-nav__link">
      Accounts and Billing
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-3-3" type="checkbox" id="nav-3-3">
    
    <label class="md-nav__link" for="nav-3-3">
      Requesting Data
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="2">
      <label class="md-nav__title" for="nav-3-3">
        Requesting Data
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../requesting-data/about-our-data/" title="About our Data" class="md-nav__link">
      About our Data
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../requesting-data/data-request/" title="Making a Data Request" class="md-nav__link">
      Making a Data Request
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../requesting-data/how-to-fill-out-DAA/" title="Filling Out a Data Access Agreement" class="md-nav__link">
      Filling Out a Data Access Agreement
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../requesting-data/glossary/" title="About our Decision Process & Terminology" class="md-nav__link">
      About our Decision Process & Terminology
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
          
          
          


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-3-4" type="checkbox" id="nav-3-4">
    
    <label class="md-nav__link" for="nav-3-4">
      Managing Data
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="2">
      <label class="md-nav__title" for="nav-3-4">
        Managing Data
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../managing-data/working-with-our-data/" title="Managing Data Overview" class="md-nav__link">
      Managing Data Overview
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../managing-data/upload-local/" title="Upload/Download Data (local)" class="md-nav__link">
      Upload/Download Data (local)
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../managing-data/upload-cluster/" title="Upload Data (HPC)" class="md-nav__link">
      Upload Data (HPC)
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../managing-data/how-to-fill-out-Extension/" title="Renewing your Data Access" class="md-nav__link">
      Renewing your Data Access
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-3-5" type="checkbox" id="nav-3-5" checked>
    
    <label class="md-nav__link" for="nav-3-5">
      Analyzing Data
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="2">
      <label class="md-nav__title" for="nav-3-5">
        Analyzing Data
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../command-line/" title="Command Line Interaction" class="md-nav__link">
      Command Line Interaction
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../running-sj-workflows/" title="Running our Workflows" class="md-nav__link">
      Running our Workflows
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-3-5-3" type="checkbox" id="nav-3-5-3" checked>
    
    <label class="md-nav__link" for="nav-3-5-3">
      Workflow Guides
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="3">
      <label class="md-nav__title" for="nav-3-5-3">
        Workflow Guides
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../chipseq/" title="ChIP-Seq Peak Calling" class="md-nav__link">
      ChIP-Seq Peak Calling
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="__toc">
    
    
      <label class="md-nav__link md-nav__link--active" for="__toc">
        cis-X
      </label>
    
    <a href="./" title="cis-X" class="md-nav__link md-nav__link--active">
      cis-X
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#overview" class="md-nav__link">
    Overview
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#inputs" class="md-nav__link">
    Inputs
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#input-file-configuration" class="md-nav__link">
    Input file configuration
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#outputs" class="md-nav__link">
    Outputs
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#creating-a-workspace" class="md-nav__link">
    Creating a workspace
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#uploading-input-files" class="md-nav__link">
    Uploading Input Files
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#running-the-workflow" class="md-nav__link">
    Running the Workflow
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#analysis-of-results" class="md-nav__link">
    Analysis of Results
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#interpreting-results" class="md-nav__link">
    Interpreting results
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#cis-activated-candidates" class="md-nav__link">
    cis-activated candidates
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sv-candidates" class="md-nav__link">
    SV candidates
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#cna-candidates" class="md-nav__link">
    CNA candidates
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#snvindel-candidates" class="md-nav__link">
    SNV/indel candidates
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ohe-results" class="md-nav__link">
    OHE results
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gene-level-ase-results" class="md-nav__link">
    Gene level ASE results
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#single-marker-ase-results" class="md-nav__link">
    Single marker ASE results
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#frequently-asked-questions" class="md-nav__link">
    Frequently asked questions
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#similar-topics" class="md-nav__link">
    Similar Topics
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
    
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../methylation-to-activity/" title="MethylationToActivity" class="md-nav__link">
      MethylationToActivity
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../mutational-signatures/" title="Mutational Signatures" class="md-nav__link">
      Mutational Signatures
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../rapid-rnaseq/" title="Rapid RNA-Seq Fusion Detection" class="md-nav__link">
      Rapid RNA-Seq Fusion Detection
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../rnaseq-expression-classification/" title="RNA-Seq Expression Classification" class="md-nav__link">
      RNA-Seq Expression Classification
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../sequencerr/" title="SequencErr" class="md-nav__link">
      SequencErr
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../warden/" title="WARDEN Differential Expression Analysis" class="md-nav__link">
      WARDEN Differential Expression Analysis
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../creating-a-cloud-app/" title="Creating your own Workflow" class="md-nav__link">
      Creating your own Workflow
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../interactive-node/" title="Interactive nodes" class="md-nav__link">
      Interactive nodes
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../visualizing-ngs-data/" title="Visualizing NGS data" class="md-nav__link">
      Visualizing NGS data
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../faq/" title="Frequently Asked Questions" class="md-nav__link">
      Frequently Asked Questions
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-4" type="checkbox" id="nav-4">
    
    <label class="md-nav__link" for="nav-4">
      PeCan
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-4">
        PeCan
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../../pecan/" title="Getting Started" class="md-nav__link">
      Getting Started
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../pecan/pecan-pie/" title="Pecan PIE" class="md-nav__link">
      Pecan PIE
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-5" type="checkbox" id="nav-5">
    
    <label class="md-nav__link" for="nav-5">
      Visualization Community
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-5">
        Visualization Community
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../../visualization-community/" title="Getting Started" class="md-nav__link">
      Getting Started
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../visualization-community/proteinpaint/" title="ProteinPaint" class="md-nav__link">
      ProteinPaint
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../visualization-community/genomepaint/" title="GenomePaint" class="md-nav__link">
      GenomePaint
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../visualization-community/faq/" title="Frequently Asked Questions" class="md-nav__link">
      Frequently Asked Questions
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-6" type="checkbox" id="nav-6">
    
    <label class="md-nav__link" for="nav-6">
      Portals
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-6">
        Portals
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../../cstn/" title="Childhood Solid Tumor Network Portal" class="md-nav__link">
      Childhood Solid Tumor Network Portal
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../pbtp/" title="Pediatric Brain Tumor Portal" class="md-nav__link">
      Pediatric Brain Tumor Portal
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-7" type="checkbox" id="nav-7">
    
    <label class="md-nav__link" for="nav-7">
      ProteinPaint
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-7">
        ProteinPaint
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-7-1" type="checkbox" id="nav-7-1">
    
    <label class="md-nav__link" for="nav-7-1">
      ProteinPaint
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="2">
      <label class="md-nav__title" for="nav-7-1">
        ProteinPaint
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../../proteinpaint/" title="Introduction" class="md-nav__link">
      Introduction
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-7-1-2" type="checkbox" id="nav-7-1-2">
    
    <label class="md-nav__link" for="nav-7-1-2">
      User Guide
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="3">
      <label class="md-nav__title" for="nav-7-1-2">
        User Guide
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../../proteinpaint/navigating-the-protein-diagram/" title="Getting Started" class="md-nav__link">
      Getting Started
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../proteinpaint/loading-datasets/" title="Loading Datasets" class="md-nav__link">
      Loading Datasets
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../proteinpaint/view-mutation-details/" title="Viewing SNV/Indel Mutations" class="md-nav__link">
      Viewing SNV/Indel Mutations
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../proteinpaint/exploring-gene-fusions/" title="Viewing Gene Fusions" class="md-nav__link">
      Viewing Gene Fusions
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../proteinpaint/exploring-gene-expression/" title="Viewing Expression Data" class="md-nav__link">
      Viewing Expression Data
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../proteinpaint/filtering-mutations/" title="Filtering Mutations" class="md-nav__link">
      Filtering Mutations
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../proteinpaint/exporting-figures-and-data/" title="Exporting Figures and Data" class="md-nav__link">
      Exporting Figures and Data
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../proteinpaint/advanced-guides/using-the-gene-table/" title="Using the Gene Table" class="md-nav__link">
      Using the Gene Table
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../proteinpaint/advanced-guides/using-the-sample-table/" title="Using the Sample Table" class="md-nav__link">
      Using the Sample Table
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../proteinpaint/advanced-guides/using-the-heatmap/" title="Using the Heatmap" class="md-nav__link">
      Using the Heatmap
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../proteinpaint/advanced-guides/using-disease-pie-charts/" title="Using Disease Pie Charts" class="md-nav__link">
      Using Disease Pie Charts
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../proteinpaint/advanced-guides/stratify-mutation-by-cancer-subtype/" title="Stratifying Mutations by Cancer Subtype" class="md-nav__link">
      Stratifying Mutations by Cancer Subtype
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
          
          
          


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-7-1-3" type="checkbox" id="nav-7-1-3">
    
    <label class="md-nav__link" for="nav-7-1-3">
      File Formats
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="3">
      <label class="md-nav__title" for="nav-7-1-3">
        File Formats
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../../proteinpaint/file-formats/snv-and-indel/" title="SNV and Indel" class="md-nav__link">
      SNV and Indel
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../proteinpaint/file-formats/sv-and-fusion/" title="SV and fusion transcript" class="md-nav__link">
      SV and fusion transcript
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../proteinpaint/file-formats/cnv/" title="Copy Number Variants" class="md-nav__link">
      Copy Number Variants
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../proteinpaint/file-formats/itd/" title="Internal tandem duplications" class="md-nav__link">
      Internal tandem duplications
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../proteinpaint/file-formats/intragenic-deletion/" title="Intragenic deletion" class="md-nav__link">
      Intragenic deletion
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../proteinpaint/file-formats/truncation/" title="Trunction, N-loss or C-loss" class="md-nav__link">
      Trunction, N-loss or C-loss
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../proteinpaint/file-formats/json-bed/" title="JSON-BED" class="md-nav__link">
      JSON-BED
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../proteinpaint/file-formats/splice-junction/" title="Splice Junction" class="md-nav__link">
      Splice Junction
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../proteinpaint/file-formats/pgv/" title="Profile-Gene-Value" class="md-nav__link">
      Profile-Gene-Value
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../proteinpaint/file-formats/bampile/" title="Bampile" class="md-nav__link">
      Bampile
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
          
          
          


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-7-1-4" type="checkbox" id="nav-7-1-4">
    
    <label class="md-nav__link" for="nav-7-1-4">
      Advanced Usage
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="3">
      <label class="md-nav__title" for="nav-7-1-4">
        Advanced Usage
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../../proteinpaint/advanced-guides/custom-protein-domains/" title="Add Custom Protein Domains" class="md-nav__link">
      Add Custom Protein Domains
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../proteinpaint/advanced-guides/using-custom-tracks/" title="Using Custom Tracks" class="md-nav__link">
      Using Custom Tracks
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../proteinpaint/advanced-guides/visualize-custom-mutations/" title="Visualize Your Mutation Data" class="md-nav__link">
      Visualize Your Mutation Data
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../proteinpaint/advanced-guides/loading-snv-indel-mutations-from-a-file/" title="Loading SNV/Indel Mutations from a File" class="md-nav__link">
      Loading SNV/Indel Mutations from a File
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../proteinpaint/advanced-guides/uploaded-mutation-data-user-interface/" title="Uploaded Mutation Data User Interface" class="md-nav__link">
      Uploaded Mutation Data User Interface
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../proteinpaint/advanced-guides/loading-tcga-maf-file/" title="Loading a TCGA MAF File" class="md-nav__link">
      Loading a TCGA MAF File
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../proteinpaint/advanced-guides/using-a-vcf-file/" title="Loading a VCF File" class="md-nav__link">
      Loading a VCF File
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
          
          
          


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-7-1-5" type="checkbox" id="nav-7-1-5">
    
    <label class="md-nav__link" for="nav-7-1-5">
      Developer's Guide
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="3">
      <label class="md-nav__title" for="nav-7-1-5">
        Developer's Guide
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../../proteinpaint/developers-guide/embedding-proteinpaint/" title="Embedding ProteinPaint" class="md-nav__link">
      Embedding ProteinPaint
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../proteinpaint/developers-guide/url-parameters/" title="URL Parameters" class="md-nav__link">
      URL Parameters
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../proteinpaint/developers-guide/organizing-data-into-a-study-view/" title="Organizing Data into a Study View" class="md-nav__link">
      Organizing Data into a Study View
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
      </ul>
    </nav>
  </li>

        
          
          
          


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-7-2" type="checkbox" id="nav-7-2">
    
    <label class="md-nav__link" for="nav-7-2">
      GenomePaint
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="2">
      <label class="md-nav__title" for="nav-7-2">
        GenomePaint
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../../genomepaint/" title="User Guide" class="md-nav__link">
      User Guide
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-8" type="checkbox" id="nav-8">
    
    <label class="md-nav__link" for="nav-8">
      Studies
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-8">
        Studies
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../../studies/" title="About" class="md-nav__link">
      About
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../studies/sickle-cell/" title="Sickle Cell" class="md-nav__link">
      Sickle Cell
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../../faq/" title="Frequently Asked Questions" class="md-nav__link">
      Frequently Asked Questions
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../../citing-stjude-cloud/" title="Citing St. Jude Cloud" class="md-nav__link">
      Citing St. Jude Cloud
    </a>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#overview" class="md-nav__link">
    Overview
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#inputs" class="md-nav__link">
    Inputs
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#input-file-configuration" class="md-nav__link">
    Input file configuration
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#outputs" class="md-nav__link">
    Outputs
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#creating-a-workspace" class="md-nav__link">
    Creating a workspace
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#uploading-input-files" class="md-nav__link">
    Uploading Input Files
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#running-the-workflow" class="md-nav__link">
    Running the Workflow
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#analysis-of-results" class="md-nav__link">
    Analysis of Results
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#interpreting-results" class="md-nav__link">
    Interpreting results
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#cis-activated-candidates" class="md-nav__link">
    cis-activated candidates
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sv-candidates" class="md-nav__link">
    SV candidates
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#cna-candidates" class="md-nav__link">
    CNA candidates
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#snvindel-candidates" class="md-nav__link">
    SNV/indel candidates
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ohe-results" class="md-nav__link">
    OHE results
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gene-level-ase-results" class="md-nav__link">
    Gene level ASE results
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#single-marker-ase-results" class="md-nav__link">
    Single marker ASE results
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#frequently-asked-questions" class="md-nav__link">
    Frequently asked questions
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#similar-topics" class="md-nav__link">
    Similar Topics
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/stjudecloud/docs/edit/master/docs/guides/genomics-platform/analyzing-data/cis-x.md" title="Edit this page" class="md-icon md-content__icon">&#xE3C9;</a>
                
                
                  <h1>cis-X</h1>
                
                <table>
<thead>
<tr>
<th></th>
<th></th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>Authors</strong></td>
<td>Yu Liu<sup>1,2,*,#</sup>, Chunliang Li<sup>3,*</sup>, Shuhong Shen<sup>1,4,*</sup>, Xiaolong Chen<sup>2</sup>, Karol Szlachta<sup>2</sup>, Michael N. Edmonson<sup>2</sup>, Ying Shao<sup>2</sup>, Xiaotu Ma<sup>2</sup>, Judith Hyle<sup>3</sup>, Shaela Wright<sup>3</sup>, Bensheng Ju<sup>2</sup>, Michael C. Rusch<sup>2</sup>, Yanling Liu<sup>2</sup>, Benshang Li<sup>1,4</sup>, Michael Macias<sup>2</sup>, Liqing Tian<sup>2</sup>, John Easton<sup>2</sup>, Maoxiang Qian<sup>5</sup>, Jun J. Yang<sup>5,6,7</sup>, Shaoyan Hu<sup>8</sup>, A. Thomas Look<sup>9,10</sup> and Jinghui Zhang<sup>2,#</sup></td>
</tr>
<tr>
<td><strong>Publication</strong></td>
<td>"Discovery of regulatory non-coding variants in individual cancer genomes using cis-X" (in submission)</td>
</tr>
<tr>
<td><strong>Technical Support</strong></td>
<td><a href="https://stjude.cloud/contact">Contact Us</a></td>
</tr>
</tbody>
</table>
<h2 id="overview">Overview<a class="headerlink" href="#overview" title="Permanent link">&para;</a></h2>
<p>Activating regular variants usually cause the cis-activation of target genes. To find cis-activated genes, allelic specific/imbalance expressions (ASE) and
outlier high expression (OHE) signals are used. Variants in the same topologically associated domains with the candidates can then be searched, including structural variants (SV), copy number aberrations (CNA), and single nucleotide variations (SNV) and insertion/deletions (indel).</p>
<p>A transcription factor binding analysis is also done, using motifs from <a href="http://hocomoco11.autosome.ru/">HOCOMOCO</a> v10 models.</p>
<p>cis-X currently only works with hg19 (GRCh37).</p>
<h2 id="inputs">Inputs<a class="headerlink" href="#inputs" title="Permanent link">&para;</a></h2>
<table>
<thead>
<tr>
<th>Name</th>
<th>Type</th>
<th>Description</th>
<th>Example</th>
</tr>
</thead>
<tbody>
<tr>
<td>Sample ID</td>
<td>String</td>
<td>The ID of the input sample</td>
<td>SJALL018373_D1</td>
</tr>
<tr>
<td>Disease subtype</td>
<td>String</td>
<td>The disease name under analysis. Must be either NBL or TALL.</td>
<td>TALL</td>
</tr>
<tr>
<td><a href="#single-nucleotide-variants">Single nucleotide variants</a></td>
<td>File</td>
<td>Tab-delimited file containing raw sequence variants</td>
<td>*.txt</td>
</tr>
<tr>
<td><a href="#cnvloh-regions">CNV/LOH regions</a></td>
<td>File</td>
<td>Tab-delimited file containing any aneuploidy region existing in the tumor genome under analysis</td>
<td>*.txt</td>
</tr>
<tr>
<td>RNA-Seq BAM</td>
<td>File</td>
<td>BAM file aligned to hg19 (GRCh37)</td>
<td>*.bam</td>
</tr>
<tr>
<td>RNA-Seq BAM index</td>
<td>File</td>
<td>BAM index for the given BAM</td>
<td>*.bam.bai</td>
</tr>
<tr>
<td><a href="#gene-expression-table">Gene expression table</a></td>
<td>File</td>
<td>Tab-delimited file containing gene level expressions for the tumor under analysis in FPKM</td>
<td>*.txt</td>
</tr>
<tr>
<td><a href="#somatic-snvindels">Somatic SNV/indels</a></td>
<td>File</td>
<td>Tab-delimited file containing somatic SNV/indels in the tumor genome</td>
<td>*.txt</td>
</tr>
<tr>
<td><a href="#somatic-svs">Somatic SVs</a></td>
<td>File</td>
<td>Tab-delimited file containing somatic acquired structural variants in the tumor genome</td>
<td>*.txt</td>
</tr>
<tr>
<td><a href="#somatic-cnvs">Somatic CNVs</a></td>
<td>File</td>
<td>Tab-delimited file containing copy number aberrations in the tumor genome</td>
<td>*.txt</td>
</tr>
<tr>
<td>CNV/LOH action</td>
<td>String</td>
<td>The behavior when handling markers in CNV/LOH regions. Can be either <code>keep</code> or <code>drop</code>. [default: keep]</td>
<td>drop</td>
</tr>
<tr>
<td>Minimum coverage for WGS</td>
<td>Integer</td>
<td>The minimum coverage in WGS to be included in the analysis [default:  10]</td>
<td>10</td>
</tr>
<tr>
<td>Minimum coverage for RNA-Seq</td>
<td>Integer</td>
<td>The minimum coverage in RNA-Seq to be included in the analysis [default: 10]</td>
<td>5</td>
</tr>
<tr>
<td>Candidate FPKM threshold</td>
<td>Float</td>
<td>The FPKM threshold for the nomination of a cis-activated candidate [default: 5.0]</td>
<td>0.1</td>
</tr>
<tr>
<td>User annotations</td>
<td>File</td>
<td>User applied annotations [optional]</td>
<td>*.bed</td>
</tr>
<tr>
<td>chr Prefix</td>
<td>String</td>
<td>Whether the names in the reference sequence dictionary are prefixed with "chr". Must be either <code>TRUE</code> or <code>FALSE</code>. [default: TRUE]</td>
<td>TRUE</td>
</tr>
<tr>
<td>TAD annotations</td>
<td>File</td>
<td>TAD annotations [optional]</td>
<td>*.bed</td>
</tr>
</tbody>
</table>
<h3 id="input-file-configuration">Input file configuration<a class="headerlink" href="#input-file-configuration" title="Permanent link">&para;</a></h3>
<p>cis-X requires six tab-delimited input files to be prepared in advance. These
files can be uploaded via the <a href="../command-line/">command line</a>.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Even though CNV/LOH regions, somatic SNV/indels, somatic SVs, and
somatic CNVs can be "empty", using such inputs will produce results with a
much higher false positive rate.</p>
</div>
<h4 id="Single nucleotide variants">Single nucleotide variants</h4>

<p>A list of single nucleotide markers is a tab-delimited file with the
following columns:</p>
<ul>
<li><code>Chr</code>: chromosome name for the marker</li>
<li><code>Pos</code>: genomic start location for the marker</li>
<li><code>Chr_Allele</code>: reference allele</li>
<li><code>Alternative_Allele</code>: alternative allele</li>
<li><code>reference_tumor_count</code>: reference allele count in the tumor genome</li>
<li><code>alternative_tumor_count</code>: alternative allele count in the tumor genome</li>
<li><code>reference_normal_count</code>: reference allele count in the matched normal genome</li>
<li><code>alternative_normal_count</code>: alternative count in the matched normal genome</li>
</ul>
<p>This file can be generated with Bambino.</p>
<h5>Example</h5>

<table>
<thead>
<tr>
<th>Chr</th>
<th align="right">Pos</th>
<th>Chr_Allele</th>
<th>Alternative_Allele</th>
<th align="right">reference_tumor_count</th>
<th align="right">alternative_tumor_count</th>
<th align="right">reference_normal_count</th>
<th>alternative_normal_count</th>
</tr>
</thead>
<tbody>
<tr>
<td>chr11</td>
<td align="right">61396</td>
<td>TT</td>
<td></td>
<td align="right">0</td>
<td align="right">3</td>
<td align="right">0</td>
<td>10</td>
</tr>
<tr>
<td>chr11</td>
<td align="right">72981</td>
<td></td>
<td>T</td>
<td align="right">1</td>
<td align="right">3</td>
<td align="right">2</td>
<td>3</td>
</tr>
</tbody>
</table>
<h4 id="CNV/LOH regions">CNV/LOH regions</h4>

<p>The CNV/LOH regions are all the genomic regions carrying copy number
variations (CNV) or loss of heterozygosity (LOH), which will be filtered out
during analysis.</p>
<p>This is a tab-delimited file in the bed format. It must have at least the
following three columns:</p>
<ul>
<li><code>chrom</code>: chromosome name</li>
<li><code>loc.start</code>: genomic start location</li>
<li><code>loc.end</code>: genomic end location</li>
</ul>
<p>If no CNV/LOH are in the genome under analysis, a file with no rows (but
including headers) can be provided.</p>
<p>This file can be generated with CONSERTING.</p>
<h5>Example</h5>

<table>
<thead>
<tr>
<th>chrom</th>
<th align="right">loc.start</th>
<th align="right">loc.end</th>
<th>Sample</th>
<th align="right">seg.mean</th>
<th align="right">LogRatio</th>
<th>source</th>
</tr>
</thead>
<tbody>
<tr>
<td>chr9</td>
<td align="right">10712</td>
<td align="right">37855747</td>
<td>SJALL018373_D1</td>
<td align="right">0.471181417</td>
<td align="right"></td>
<td>LOH</td>
</tr>
<tr>
<td>chr9</td>
<td align="right">20276901</td>
<td align="right">20703900</td>
<td>SJALL018373_D1</td>
<td align="right">-0.978</td>
<td align="right">-5.696</td>
<td>CNV</td>
</tr>
</tbody>
</table>
<h4 id="Gene expression table">Gene expression table</h4>

<p>The gene expression table is a tab-delimited file containing gene level
expressions for the tumor under analysis. The expressions are in FPKM
(fragments per kilobase of transcript per million mapped reads).</p>
<ul>
<li><code>GeneID</code>: gene <a href="http://www.ensembl.org/">Ensembl</a> ID</li>
<li><code>GeneName</code>: gene symbol</li>
<li><code>Type</code>: <a href="https://www.gencodegenes.org/gencode_biotypes.html">transcript type</a></li>
<li><code>Status</code>: transcript status (must be <code>KNOWN</code>, <code>NOVEL</code>, or <code>PUTATIVE</code>)</li>
<li><code>Chr</code>: chromosome name</li>
<li><code>Start</code> genomic start location</li>
<li><code>End</code>: genomic end location</li>
<li>[SampleID...]: FPKM for the given sample</li>
</ul>
<p>This file can be generated with the output of HTseq-count preprocessed
through <code>mergeData_geneName.pl</code> (available with the distribution of cis-X).
The data must be able to match values in the given gene specific reference
expression matrices generated from a larger cohort.</p>
<h5>Example</h5>

<table>
<thead>
<tr>
<th>GeneID</th>
<th>GeneName</th>
<th>Type</th>
<th>Status</th>
<th>Chr</th>
<th align="right">Start</th>
<th align="right">End</th>
<th align="right">SJALL018373_D1</th>
</tr>
</thead>
<tbody>
<tr>
<td>ENSG00000261122.2</td>
<td>5S_rRNA</td>
<td>lincRNA</td>
<td>NOVEL</td>
<td>chr16</td>
<td align="right">34977639</td>
<td align="right">34990886</td>
<td align="right">0.0000</td>
</tr>
<tr>
<td>ENSG00000249352.3</td>
<td>7SK</td>
<td>lincRNA</td>
<td>NOVEL</td>
<td>chr5</td>
<td align="right">68266266</td>
<td align="right">68325992</td>
<td align="right">4.5937</td>
</tr>
</tbody>
</table>
<h4 id="Somatic SNV/indels">Somatic SNV/indels</h4>

<p>This is a tab-delimited file containing somatic sequence mutations present in
the genome under analysis. It includes both single nucleotide variants (SNV)
and small insertion/deletions (indel). The file must have the following
columns:</p>
<ul>
<li><code>chr</code>: chromosome name</li>
<li><code>pos</code>: genomic start location</li>
<li><code>ref</code>: reference nucleotide</li>
<li><code>mutant</code>: mutant nucleotide</li>
<li><code>type</code>: mutation type (must be either <code>snv</code> or <code>indel</code>)</li>
</ul>
<p>Note that the coordinate used for an indel is after the inserted sequence.</p>
<p>If no SNV/indels are in the sample under analysis, a file with no rows
(but including headers) can be provided.</p>
<p>This file can can be created with Bambino and then preprocessed using the
steps taken in "<a href="https://www.ncbi.nlm.nih.gov/pubmed/22237106">The genetic basis of early T-cell precursor acute lymphoblastic leukaemia</a>".</p>
<h5>Example</h5>

<table>
<thead>
<tr>
<th>chr</th>
<th align="right">pos</th>
<th>ref</th>
<th>mut</th>
<th>type</th>
</tr>
</thead>
<tbody>
<tr>
<td>chr1</td>
<td align="right">24782720</td>
<td>G</td>
<td>A</td>
<td>snv</td>
</tr>
<tr>
<td>chr11</td>
<td align="right">82896176</td>
<td>T</td>
<td>C</td>
<td>snv</td>
</tr>
</tbody>
</table>
<h4 id="Somatic SVs">Somatic SVs</h4>

<p>This is a tab-delimited file containing somatic-acquired structural variants
(SV) in the cancer genome. The file must have the following columns:</p>
<ul>
<li><code>chrA</code>: chromosome name of the left breakpoint</li>
<li><code>posA</code>: genomic location of the left breakpoint</li>
<li><code>ortA</code>: strand orientation of the left breakpoint</li>
<li><code>chrB</code>: chromosome name of the right breakpoint</li>
<li><code>posB</code>: genomic location of the right breakpoint</li>
<li><code>ortB</code>: strand orientation of the right breakpoint</li>
</ul>
<p>Strand orientations are denoted with a <code>+</code> for a sense or coding strand and
<code>-</code> for a antisense or non-coding strand.</p>
<p>If no somatic SVs are in the sample under analysis, a file with no rows (but
including headers) can be provided.</p>
<p>This file can be generated by CREST.</p>
<h5>Example</h5>

<table>
<thead>
<tr>
<th>chrA</th>
<th align="right">posA</th>
<th>ortA</th>
<th>chrB</th>
<th align="right">posB</th>
<th>ortB</th>
<th>type</th>
</tr>
</thead>
<tbody>
<tr>
<td>chr11</td>
<td align="right">33913169</td>
<td>+</td>
<td>chr7</td>
<td align="right">142494049</td>
<td>-</td>
<td>CTX</td>
</tr>
<tr>
<td>chr11</td>
<td align="right">64219334</td>
<td>+</td>
<td>chr2</td>
<td align="right">205042527</td>
<td>-</td>
<td>CTX</td>
</tr>
</tbody>
</table>
<h4 id="Somatic CNVs">Somatic CNVs</h4>

<p>This is a tab-delimited file containing the genomic
regions with somatic-acquired copy number aberrations (CNA) in the cancer
genome.</p>
<ul>
<li><code>chr</code>: chromosome name</li>
<li><code>start</code>: genomic start location</li>
<li><code>end</code>: genomic end location</li>
<li><code>logR</code>: log2 ratio</li>
</ul>
<p>If no somatic CNVs are in the sample under analysis, a file with no rows
(but including headers) can be provided.</p>
<p>This file can be generating by CONSERTING.</p>
<h5>Example</h5>

<table>
<thead>
<tr>
<th>chr</th>
<th align="right">start</th>
<th align="right">end</th>
<th align="right">logR</th>
</tr>
</thead>
<tbody>
<tr>
<td>chr9</td>
<td align="right">20276901</td>
<td align="right">20703900</td>
<td align="right">-5.696</td>
</tr>
</tbody>
</table>
<h2 id="outputs">Outputs<a class="headerlink" href="#outputs" title="Permanent link">&para;</a></h2>
<table>
<thead>
<tr>
<th>Name</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="#cis-activated-candidates">cis-activated candidates</a></td>
<td>cis-activated candidates in the tumor genome under analysis</td>
</tr>
<tr>
<td><a href="#sv-candidates">SV candidates</a></td>
<td>Structural variant (SV) candidates predicted as the causal for the cis-activated genes in the regulatory territory</td>
</tr>
<tr>
<td><a href="#cna-candidates">CNA candidates</a></td>
<td>Copy number aberrations (CNA) predicted as the causal for the cis-activated genes in the regulatory territory</td>
</tr>
<tr>
<td><a href="#snvindel-candidates">SNV/indel candidates</a></td>
<td>SNV/indel candidates predicted as functional and predicted transcription factors</td>
</tr>
<tr>
<td><a href="#ohe-results">OHE results</a></td>
<td>Raw outlier high expression (OHE) results</td>
</tr>
<tr>
<td><a href="#gene-level-ase-results">Gene level ASE results</a></td>
<td>Raw gene level allelic specific expression (ASE) results</td>
</tr>
<tr>
<td><a href="#single-marker-ase-results">Single marker ASE results</a></td>
<td>Raw single marker allelic specific expression (ASE) results</td>
</tr>
</tbody>
</table>
<h2 id="creating-a-workspace">Creating a workspace<a class="headerlink" href="#creating-a-workspace" title="Permanent link">&para;</a></h2>
<p>Before you can run one of our workflows, you must first create a workspace in DNAnexus for the run. Refer to <a href="../running-sj-workflows/#getting-started">the general workflow guide</a> to learn how to create a DNAnexus workspace for each workflow run.</p>
<p>You can navigate to the Cis-X workflow page <a href="https://platform.stjude.cloud/workflows/cis-x">here</a></p>
<h2 id="uploading-input-files">Uploading Input Files<a class="headerlink" href="#uploading-input-files" title="Permanent link">&para;</a></h2>
<p>cis-X requires a total of eight files to be uploaded, as <a href="#input-file-configuration">input</a>. </p>
<p>Refer to <a href="../running-sj-workflows/#uploading-files">the general workflow guide</a> to learn how to upload input files to the workspace you just created.</p>
<h2 id="running-the-workflow">Running the Workflow<a class="headerlink" href="#running-the-workflow" title="Permanent link">&para;</a></h2>
<p>Refer to <a href="../running-sj-workflows/#running-the-workflow">the general workflow guide</a> to learn how to launch the workflow, hook up input files, adjust parameters, start a run, and monitor run progress.</p>
<h2 id="analysis-of-results">Analysis of Results<a class="headerlink" href="#analysis-of-results" title="Permanent link">&para;</a></h2>
<p>Each tool in St. Jude Cloud produces a visualization that makes understanding results more accessible than working with excel spreadsheet or tab delimited files. This is the primary way we recommend you work with your results. </p>
<p>Refer to <a href="../running-sj-workflows/#custom-visualizations">the general workflow guide</a> to learn how to access these visualizations.</p>
<p>We also include the raw output files for you to dig into if the visualization is not sufficient to answer your research question.</p>
<p>Refer to <a href="../running-sj-workflows/#raw-results-files">the general workflow guide</a> to learn how to access raw results files.</p>
<h2 id="interpreting-results">Interpreting results<a class="headerlink" href="#interpreting-results" title="Permanent link">&para;</a></h2>
<h3 id="cis-activated-candidates">cis-activated candidates<a class="headerlink" href="#cis-activated-candidates" title="Permanent link">&para;</a></h3>
<p>The main result file contains the cis-activated candidates in the tumor genome
under analysis.</p>
<ul>
<li><code>gene</code>: gene accession number (<a href="https://www.ncbi.nlm.nih.gov/refseq/">RefSeq</a> ID)</li>
<li><code>gsym</code>: gene symbol</li>
<li><code>chrom</code>: chromosome name</li>
<li><code>strand</code>: strand orientation</li>
<li><code>start</code>: genomic start location</li>
<li><code>end</code>: genomic end location</li>
<li><code>cdsStartStat</code>: coding sequence (CDS) start status</li>
<li><code>cdsEndStat</code>: coding sequence (CDS) end status</li>
<li><code>markers</code>: number of heterozygous markers in this gene</li>
<li><code>ase_markers</code>: number of heterozygous markers showing allelic specific expressions (ASE)</li>
<li><code>average_ai_all</code>: average B-allele frequency (BAF) difference between RNA and DNA for all heterozygous markers</li>
<li><code>average_ai_ase</code>: average BAF difference between RNA and DNA for ASE markers</li>
<li><code>pval_all_markers</code>: p-value for each marker in the ASE test</li>
<li><code>pval_ase_markers</code>: p-value for ASE markers in the ASE test</li>
<li><code>ai_all_markers</code>: BAF difference between RNA and DNA for all heterozygrous markers</li>
<li><code>ai_ase_markers</code>: BAF difference between RNA and DNA for ASE markers</li>
<li><code>comb.pval</code>: combined p-value for the ASE test</li>
<li><code>mean.delta</code>: average BAF difference between RNA and DNA for all markers</li>
<li><code>rawp</code>: raw p-value for the ASE test</li>
<li><code>Bonferroni</code>: adjusted p-value for the ASE test (single-step Bonferroni)</li>
<li><code>ABH</code>: adjusted p-value for the ASE test (Benjamini-Hochberg)</li>
<li><code>FPKM</code>: FPKM value</li>
<li><code>loo.source</code>: which reference expression matrix was used in the outlier high expression (OHE) test</li>
<li><code>loo.cohort.size</code>: number of cases in the reference expression matrix for this gene</li>
<li><code>loo.pval</code>: p-value of the OHE test</li>
<li><code>loo.rank</code>: rank of the case under analysis among the reference cases</li>
<li><code>imprinting.status</code>: imprinting status of the gene</li>
<li><code>candidate.group</code>: status of the gene, combining both ASE and outlier tests</li>
</ul>
<p>Strand orientations are denoted with a <code>+</code> for a sense or coding strand and
<code>-</code> for a antisense or non-coding strand.</p>
<p>Coding sequence status is typically one of "none" (not specified), "unk"
(unknown), "incmpl" (incomplete), or "cmpl" (complete).</p>
<h5>Example</h5>

<table>
<thead>
<tr>
<th>gene</th>
<th>gsym</th>
<th>chrom</th>
<th>strand</th>
<th align="right">start</th>
<th align="right">end</th>
<th>cdsStartStat</th>
<th>cdsEndStat</th>
<th align="right">markers</th>
<th align="right">ase_markers</th>
<th align="right">average_ai_all</th>
<th align="right">average_ai_ase</th>
<th>pval_all_markers</th>
<th>pval_ase_markers</th>
<th>ai_all_markers</th>
<th>ai_ase_markers</th>
<th align="right">comb.pval</th>
<th align="right">mean.delta</th>
<th align="right">rawp</th>
<th align="right">Bonferroni</th>
<th align="right">ABH</th>
<th align="right">FPKM</th>
<th>loo.source</th>
<th align="right">loo.cohort.size</th>
<th align="right">loo.pval</th>
<th align="right">loo.rank</th>
<th>imprinting.status</th>
<th>candidate.group</th>
</tr>
</thead>
<tbody>
<tr>
<td>NM_145804</td>
<td>ABTB2</td>
<td>chr11</td>
<td>-</td>
<td align="right">34172533</td>
<td align="right">34379555</td>
<td>cmpl</td>
<td>cmpl</td>
<td align="right">5</td>
<td align="right">5</td>
<td align="right">0.5</td>
<td align="right">0.500</td>
<td>0.001953125,0.001953125,0.001953125,6.10351562500001e-05,0.000244140625</td>
<td>0.001953125,0.001953125,0.001953125,6.10351562500001e-05,0.000244140625</td>
<td>0.5,0.5,0.5,0.5,0.5</td>
<td>0.5,0.5,0.5,0.5,0.5</td>
<td align="right">0.000644290972057077</td>
<td align="right">0.5</td>
<td align="right">0.000644290972057077</td>
<td align="right">0.632049443587993</td>
<td align="right">0.0110866672927557</td>
<td align="right">7.6776</td>
<td>bi_cohort</td>
<td align="right">40</td>
<td align="right">0.0367241086505276</td>
<td align="right">1</td>
<td></td>
<td>ase_outlier</td>
</tr>
<tr>
<td>NM_003189</td>
<td>TAL1</td>
<td>chr1</td>
<td>-</td>
<td align="right">47681961</td>
<td align="right">47698007</td>
<td>cmpl</td>
<td>cmpl</td>
<td align="right">2</td>
<td align="right">2</td>
<td align="right">0.482</td>
<td align="right">0.482</td>
<td>6.66361745922277e-28,3.30872245021211e-24</td>
<td>6.66361745922277e-28,3.30872245021211e-24</td>
<td>0.464912280701754,0.5</td>
<td>0.464912280701754,0.5</td>
<td align="right">4.69553625126628e-26</td>
<td align="right">0.482456140350877</td>
<td align="right">4.69553625126628e-26</td>
<td align="right">4.60632106249222e-23</td>
<td align="right">6.11761294450693e-24</td>
<td align="right">8.8168</td>
<td>white_list</td>
<td align="right">167</td>
<td align="right">0.0139385771987089</td>
<td align="right">1</td>
<td></td>
<td>ase_outlier</td>
</tr>
</tbody>
</table>
<h3 id="sv-candidates">SV candidates<a class="headerlink" href="#sv-candidates" title="Permanent link">&para;</a></h3>
<p>Structural variant (SV) candidates include candidates predicted as the causal
for the cis-activated genes in the regulatory territory.</p>
<ul>
<li><code>left.candidate.inTAD</code>: cis-activated candidate near the left breakpoint</li>
<li><code>right.candidate.inTAD</code>: cis-activated candidate near the right breakpoint</li>
<li><code>chrA</code>: chromosome name of the left breakpoint</li>
<li><code>posA</code>: genomic location of the left breakpoint</li>
<li><code>ortA</code>: strand orientation of the left breakpoint</li>
<li><code>chrB</code>: chromosome name of the right breakpoint</li>
<li><code>posB</code>: genomic location of the right breakpoint</li>
<li><code>ortB</code>: strand orientation of the right breakpoint</li>
<li><code>type</code>: type of translocation</li>
</ul>
<h5>Example</h5>

<table>
<thead>
<tr>
<th>left.candidate.inTAD</th>
<th>right.candidate.inTAD</th>
<th>chrA</th>
<th align="right">posA</th>
<th>ortA</th>
<th>chrB</th>
<th align="right">posB</th>
<th>ortB</th>
<th>type</th>
</tr>
</thead>
<tbody>
<tr>
<td>LMO2</td>
<td></td>
<td>chr11</td>
<td align="right">33913169</td>
<td>+</td>
<td>chr7</td>
<td align="right">142494049</td>
<td>-</td>
<td>CTX</td>
</tr>
</tbody>
</table>
<h3 id="cna-candidates">CNA candidates<a class="headerlink" href="#cna-candidates" title="Permanent link">&para;</a></h3>
<p>Copy number aberration (CNA) candidates include candidates predicted as the
causal for the cis-activated genes in the regulatory territory.</p>
<ul>
<li><code>candidate.inTAD</code>: cis-activated candidate by the CNA</li>
<li><code>chr</code>: chromosome name</li>
<li><code>start</code>: genomic start position</li>
<li><code>end</code>: genomic end location</li>
<li><code>logR</code>: log ratio of the CNA</li>
</ul>
<h3 id="snvindel-candidates">SNV/indel candidates<a class="headerlink" href="#snvindel-candidates" title="Permanent link">&para;</a></h3>
<p>SNV/indel candidates include predicted candidates as functional and predicted
transcription factors. The mutations are also annotated for known regulatory
elements reported by the <a href="https://egg2.wustl.edu/roadmap/web_portal/index.html">NIH Roadmap Epigenomics Project</a> by collecting 111
cell lines.</p>
<ul>
<li><code>chrom</code>: chromosome name</li>
<li><code>pos</code>: genomic start position</li>
<li><code>ref</code>: reference allele genotype</li>
<li><code>mut</code>: mutant allele genotype</li>
<li><code>type</code>: mutation type (either <code>snv</code> or <code>indel</code>)</li>
<li><code>target</code>: cis-activated candidate</li>
<li><code>dist</code>: distance between the mutation and transcription start sites of the target gene</li>
<li><code>tf</code>: transcription factors predicted to have the binding motif introduced by the mutation</li>
<li><code>EpiRoadmap_enhancer</code>: enhancer regions that overlap with the mutation (from the <a href="https://egg2.wustl.edu/roadmap/web_portal/index.html">NIH Roadmap Epigenomics Project</a>)</li>
<li><code>EpiRoadmap_promoter</code>: promoter regions that overlap with the mutation (from the <a href="https://egg2.wustl.edu/roadmap/web_portal/index.html">NIH Roadmap Epigenomics Project</a>)</li>
<li><code>EpiRoadmap_dyadic</code>: dyadic regions that overlap with the mutation (from the <a href="https://egg2.wustl.edu/roadmap/web_portal/index.html">NIH Roadmap Epigenomics Project</a>)</li>
</ul>
<h5>Example</h5>

<table>
<thead>
<tr>
<th>chrom</th>
<th align="right">pos</th>
<th>ref</th>
<th>mut</th>
<th>type</th>
<th>target</th>
<th>dist</th>
<th>tf</th>
<th>EpiRoadmap_enhancer</th>
<th>EpiRoadmap_promoter</th>
<th>EpiRoadmap_dyadic</th>
</tr>
</thead>
<tbody>
<tr>
<td>chr1</td>
<td align="right">47696311</td>
<td>C</td>
<td>T</td>
<td>snv</td>
<td>TAL1</td>
<td>1696</td>
<td>BCL11A,CEBPG,PBX2,YY1,ZBTB4</td>
<td></td>
<td>Brain,Digestive,ES-deriv,ESC,HSC &amp; B-cell,Heart,Muscle,Other,Sm. Muscle,iPSC</td>
<td></td>
</tr>
</tbody>
</table>
<h3 id="ohe-results">OHE results<a class="headerlink" href="#ohe-results" title="Permanent link">&para;</a></h3>
<p>OHE results are the raw results for the outlier expression test.</p>
<ul>
<li><code>Gene</code>: gene symbol</li>
<li><code>fpkm.raw</code>: FPKM value</li>
<li><code>size.bi</code>: number of cases in the bi-allelic reference cohort</li>
<li><code>p.bi</code>: p-value in the outlier test using the bi-allelic reference cohort</li>
<li><code>rank.bi</code>: rank of the expression level in the case under analysis compared to the bi-allelic reference cohort</li>
<li><code>size.cohort</code>: number of cases in the entire reference cohort</li>
<li><code>p.cohort</code>: p-value in the outlier test using the entire reference cohort</li>
<li><code>rank.cohort</code>: rank of the expression level in the case under analysis compared to the entire reference cohort</li>
<li><code>size.white</code>: number of cases in the whitelist reference cohort</li>
<li><code>p.white</code>: p-value in the outlier test using the whitelist reference cohort</li>
<li><code>rank.white</code>: rank of the expression level in the case under analysis compared to the whitelist reference cohort</li>
</ul>
<h5>Example</h5>

<table>
<thead>
<tr>
<th>Gene</th>
<th align="right">fpkm.raw</th>
<th align="right">size.bi</th>
<th align="right">p.bi</th>
<th align="right">rank.bi</th>
<th align="right">size.cohort</th>
<th align="right">p.cohort</th>
<th align="right">rank.cohort</th>
<th align="right">size.white</th>
<th align="right">p.white</th>
<th align="right">rank.white</th>
</tr>
</thead>
<tbody>
<tr>
<td>7SK</td>
<td align="right">4.5937</td>
<td align="right">na</td>
<td align="right">na</td>
<td align="right">na</td>
<td align="right">264</td>
<td align="right">0.716284011918374</td>
<td align="right">162</td>
<td align="right">na</td>
<td align="right">na</td>
<td align="right">na</td>
</tr>
<tr>
<td>A1BG</td>
<td align="right">0.2312</td>
<td align="right">24</td>
<td align="right">0.900132642257996</td>
<td align="right">21</td>
<td align="right">264</td>
<td align="right">0.84055666600945</td>
<td align="right">222</td>
<td align="right">na</td>
<td align="right">na</td>
<td align="right">na</td>
</tr>
</tbody>
</table>
<h3 id="gene-level-ase-results">Gene level ASE results<a class="headerlink" href="#gene-level-ase-results" title="Permanent link">&para;</a></h3>
<p>Gene level ASE results are the raw results from the gene level ASE test.</p>
<ul>
<li><code>gene</code>: gene accession number (<a href="https://www.ncbi.nlm.nih.gov/refseq/">RefSeq</a> ID)</li>
<li><code>gsym</code>: gene symbol</li>
<li><code>chrom</code>: chromosome name</li>
<li><code>strand</code>: strand orientation</li>
<li><code>start</code>: genomic start location</li>
<li><code>end</code>: genomic end location</li>
<li><code>cdsStartStat</code>: coding sequence (CDS) start status</li>
<li><code>cdsEndStat</code>: coding sequence (CDS) end status</li>
<li><code>markers</code>: number of heterozygous markers in this gene</li>
<li><code>ase_markers</code>: number of heterozygous markers showing allelic specific expressions (ASE)</li>
<li><code>average_ai_all</code>: average B-allele frequency (BAF) difference between RNA and DNA for all heterozygous markers</li>
<li><code>average_ai_ase</code>: average BAF difference between RNA and DNA for ASE markers</li>
<li><code>pval_all_markers</code>: p-value for each marker in the ASE test</li>
<li><code>pval_ase_markers</code>: p-value for ASE markers in the ASE test</li>
<li><code>ai_all_markers</code>: BAF difference between RNA and DNA for all heterozygrous markers</li>
<li><code>ai_ase_markers</code>: BAF difference between RNA and DNA for ASE markers</li>
<li><code>comb.pval</code>: combined p-value for the ASE test</li>
<li><code>mean.delta</code>: average BAF difference between RNA and DNA for all markers</li>
<li><code>rawp</code>: raw p-value for the ASE test</li>
<li><code>Bonferroni</code>: adjusted p-value for the ASE test (single-step Bonferroni)</li>
<li><code>ABH</code>: adjusted p-value for the ASE test (Benjamini-Hochberg)</li>
</ul>
<h5>Example</h5>

<table>
<thead>
<tr>
<th>gene</th>
<th>gsym</th>
<th>chrom</th>
<th>strand</th>
<th align="right">start</th>
<th align="right">end</th>
<th>cdsStartStat</th>
<th>cdsEndStat</th>
<th align="right">markers</th>
<th align="right">ase_markers</th>
<th align="right">average_ai_all</th>
<th align="right">average_ai_ase</th>
<th>pval_all_markers</th>
<th>pval_ase_markers</th>
<th>ai_all_markers</th>
<th>ai_ase_markers</th>
<th align="right">comb.pval</th>
<th align="right">mean.delta</th>
<th align="right">rawp</th>
<th align="right">Bonferroni</th>
<th align="right">ABH</th>
</tr>
</thead>
<tbody>
<tr>
<td>NM_024684</td>
<td>AAMDC</td>
<td>chr11</td>
<td>+</td>
<td align="right">77532207</td>
<td align="right">77583398</td>
<td>cmpl</td>
<td>cmpl</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0.079</td>
<td align="right">na</td>
<td>0.924775093657227,0.0331439677875056</td>
<td>na</td>
<td>0.00892857142857145,0.149122807017544</td>
<td>na</td>
<td align="right">0.175073458624837</td>
<td align="right">0.0790256892230577</td>
<td align="right">0.175073458624837</td>
<td align="right">1</td>
<td align="right">0.480780882445856</td>
</tr>
<tr>
<td>NM_015423</td>
<td>AASDHPPT</td>
<td>chr11</td>
<td>+</td>
<td align="right">105948291</td>
<td align="right">105969419</td>
<td>cmpl</td>
<td>cmpl</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0.023</td>
<td align="right">na</td>
<td>0.749258624760841,1</td>
<td>na</td>
<td>0.0384615384615384,0.00769230769230766</td>
<td>na</td>
<td align="right">0.86559726476049</td>
<td align="right">0.023076923076923</td>
<td align="right">0.86559726476049</td>
<td align="right">1</td>
<td align="right">0.873257417545981</td>
</tr>
</tbody>
</table>
<h3 id="single-marker-ase-results">Single marker ASE results<a class="headerlink" href="#single-marker-ase-results" title="Permanent link">&para;</a></h3>
<p>Single marker ASE results are the raw results from the single marker ASE test.</p>
<ul>
<li><code>chrom</code>: chromosome name</li>
<li><code>pos</code>: genomic start position</li>
<li><code>ref</code>: reference allele genotype</li>
<li><code>mut</code>: non-reference allele genotype</li>
<li><code>cvg_wgs</code>: coverage of the marker from the whole genome sequence (WGS)</li>
<li><code>mut_freq_wgs</code>: non-reference allele fraction in the WGS</li>
<li><code>cvg_rna</code>: coverage of the marker from the RNA-Seq</li>
<li><code>mut_freq_rna</code>: non-reference allele fraction in the RNA-Seq</li>
<li><code>ref.1</code>: read count of the reference allele in the RNA-Seq</li>
<li><code>var</code>: read count of the non-reference allele in the RNA-Seq</li>
<li><code>pvalue</code>: p-value from the binomial test</li>
<li><code>delta.abs</code>: absolute difference of the non-reference allele fraction between the WGS and RNA-Seq</li>
</ul>
<h5>Example</h5>

<table>
<thead>
<tr>
<th>chrom</th>
<th align="right">pos</th>
<th>ref</th>
<th>mut</th>
<th align="right">cvg_wgs</th>
<th align="right">mut_freq_wgs</th>
<th align="right">cvg_rna</th>
<th align="right">mut_freq_rna</th>
<th align="right">ref.1</th>
<th align="right">var</th>
<th align="right">pvalue</th>
<th align="right">delta.abs</th>
</tr>
</thead>
<tbody>
<tr>
<td>chr11</td>
<td align="right">204147</td>
<td>G</td>
<td>A</td>
<td align="right">36</td>
<td align="right">0.472</td>
<td align="right">85</td>
<td align="right">0.553</td>
<td align="right">38</td>
<td align="right">47</td>
<td align="right">0.385669420119278</td>
<td align="right">0.0529411764705883</td>
</tr>
<tr>
<td>chr11</td>
<td align="right">205198</td>
<td>C</td>
<td>A</td>
<td align="right">23</td>
<td align="right">0.522</td>
<td align="right">83</td>
<td align="right">0.313</td>
<td align="right">57</td>
<td align="right">26</td>
<td align="right">0.000877551780002863</td>
<td align="right">0.186746987951807</td>
</tr>
</tbody>
</table>
<h2 id="frequently-asked-questions">Frequently asked questions<a class="headerlink" href="#frequently-asked-questions" title="Permanent link">&para;</a></h2>
<p>None yet! If you have any questions not covered here, feel free to reach
out on <a href="https://hospital.stjude.org/apps/forms/fb/st-jude-cloud-contact/">our contact form</a>.</p>
<h2 id="similar-topics">Similar Topics<a class="headerlink" href="#similar-topics" title="Permanent link">&para;</a></h2>
<p><a href="../running-sj-workflows/">Running our Workflows</a><br />
<a href="../../managing-data/working-with-our-data/">Working with our Data Overview</a> <br />
<a href="../../managing-data/upload-local/">Upload/Download Data (local)</a>   </p>
<h2 id="Footnotes">Footnotes</h2>

<p><small></p>
<p><sup>1</sup> Pediatric Translational Medicine Institute, Shanghai Children's
  Medical Center, Shanghai Jiao Tong University School of Medicine, Shanghai,
  China</p>
<p><sup>2</sup> Department of Computational Biology, St. Jude Children's
  Research Hospital, Memphis, TN 38105, USA</p>
<p><sup>3</sup> Department of Tumor Cell Biology, St. Jude Children's Research
  Hospital, Memphis, TN 38105, USA</p>
<p><sup>4</sup> Key Laboratory of Pediatric Hematology &amp; Oncology Ministry of
  Health, Department of Hematology &amp; Oncology, Shanghai Children's Medical
  Center, Shanghai Jiao Tong University School of Medicine, Shanghai, China</p>
<p><sup>5</sup> Department of Pharmaceutical Sciences, St. Jude Children's
  Research Hospital, Memphis, TN 38105, USA</p>
<p><sup>6</sup> Hematological Malignancies Program, St. Jude Children's Research
  Hospital, Memphis, TN 38105, USA</p>
<p><sup>7</sup> Department of Oncology, St. Jude Children's Research Hospital,
  Memphis, TN 38105, USA</p>
<p><sup>8</sup> Children's Hospital of Soochow University, Suzhou, Jiangsu,
  China</p>
<p><sup>9</sup> Department of Pediatric Oncology, Dana-Farber Cancer Institute,
  Harvard Medical School, Boston, MA 02215, USA</p>
<p><sup>10</sup> Division of Pediatric Hematology-Oncology, Boston Children's
  Hospital, MA 02115, USA</p>
<p><sup>*</sup> Contributed equally to this work.</p>
<p><sup>#</sup> Correspondence should be addressed to Y.L. (<a href="&#109;&#97;&#105;&#108;&#116;&#111;&#58;&#108;&#105;&#117;&#121;&#117;&#64;&#115;&#99;&#109;&#99;&#46;&#99;&#111;&#109;&#46;&#99;&#110;">&#108;&#105;&#117;&#121;&#117;&#64;&#115;&#99;&#109;&#99;&#46;&#99;&#111;&#109;&#46;&#99;&#110;</a>)
  or J.Z. (<a href="&#109;&#97;&#105;&#108;&#116;&#111;&#58;&#106;&#105;&#110;&#103;&#104;&#117;&#105;&#46;&#122;&#104;&#97;&#110;&#103;&#64;&#115;&#116;&#106;&#117;&#100;&#101;&#46;&#111;&#114;&#103;">&#106;&#105;&#110;&#103;&#104;&#117;&#105;&#46;&#122;&#104;&#97;&#110;&#103;&#64;&#115;&#116;&#106;&#117;&#100;&#101;&#46;&#111;&#114;&#103;</a>).</p>
<p></small></p>
                
                  
                
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../chipseq/" title="ChIP-Seq Peak Calling" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                ChIP-Seq Peak Calling
              </span>
            </div>
          </a>
        
        
          <a href="../methylation-to-activity/" title="MethylationToActivity" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                MethylationToActivity
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            Copyright &copy; 2020 St. Jude Children's Research Hospital
          </div>
        
        
        built from <a href="https://github.com/stjudecloud/docs/tree/eba8e43">eba8e43</a> using
        
        <a href="https://www.mkdocs.org">MkDocs</a>
        and a modified version of
        <a href="https://squidfunk.github.io/mkdocs-material/">
          Material for MkDocs
        </a>.
      </div>
      
  <div class="md-footer-social">
    <link rel="stylesheet" href="../../../../assets/fonts/font-awesome.css">
    
      <a href="https://www.facebook.com/stjuderesearch/" target="_blank" rel="noopener" title="facebook" class="md-footer-social__link fa fa-facebook"></a>
    
      <a href="https://twitter.com/stjuderesearch?lang=en" target="_blank" rel="noopener" title="twitter" class="md-footer-social__link fa fa-twitter"></a>
    
      <a href="https://www.linkedin.com/company/st-jude-childrens-research-hospital" target="_blank" rel="noopener" title="linkedin" class="md-footer-social__link fa fa-linkedin"></a>
    
      <a href="https://github.com/stjude" target="_blank" rel="noopener" title="github" class="md-footer-social__link fa fa-github"></a>
    
  </div>

    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../../../assets/javascripts/application.2d55c750.js"></script>
      
      <script>app.initialize({version:"1.1",url:{base:"../../../.."}})</script>
      
    
  </body>
</html>