



<!doctype html>
<html lang="en" class="no-js">
  <head>
    <script src="//assets.adobedtm.com/launch-EN942f79c83ff84c10a45b4856a85eaccb.min.js" async></script>
    <script async src="https://www.googletagmanager.com/gtag/js?id=UA-107924368-1"></script>
    <script>function gtag(){dataLayer.push(arguments)}window.dataLayer=window.dataLayer||[],gtag("js",new Date),gtag("config","UA-107924368-1")</script>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
      
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.pipeline.stopwords" content="True">
      
        <meta name="lang:search.pipeline.trimmer" content="True">
      
        <meta name="lang:search.result.none" content="No matching documents">
      
        <meta name="lang:search.result.one" content="1 matching document">
      
        <meta name="lang:search.result.other" content="# matching documents">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="../../../images/favicon.ico">
      <meta name="generator" content="mkdocs-1.1, mkdocs-material-4.6.3">
    
    
      
        <title>NeoepitopePred - St. Jude Cloud Documentation</title>
      
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/application.7d2ef1ff.css">
      
      
    
    
      <script src="../../../assets/javascripts/modernizr.eca31fed.js"></script>
    
    
      
    
    <link rel="stylesheet" href="../../../assets/fonts/material-icons.css">
    
    
      <link rel="stylesheet" href="../../../css/extra.css">
    
    
      
    
    
  </head>
  
    <body dir="ltr">
  
    <svg class="md-svg">
      <defs>
        
        
          <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448" viewBox="0 0 416 448" id="__github"><path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19T128 352t-18.125-8.5-10.75-19T96 304t3.125-20.5 10.75-19T128 256t18.125 8.5 10.75 19T160 304zm160 0q0 10-3.125 20.5t-10.75 19T288 352t-18.125-8.5-10.75-19T256 304t3.125-20.5 10.75-19T288 256t18.125 8.5 10.75 19T320 304zm40 0q0-30-17.25-51T296 232q-10.25 0-48.75 5.25Q229.5 240 208 240t-39.25-2.75Q130.75 232 120 232q-29.5 0-46.75 21T56 304q0 22 8 38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0 37.25-1.75t35-7.375 30.5-15 20.25-25.75T360 304zm56-44q0 51.75-15.25 82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5T212 416q-19.5 0-35.5-.75t-36.875-3.125-38.125-7.5-34.25-12.875T37 371.5t-21.5-28.75Q0 312 0 260q0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25 30.875Q171.5 96 212 96q37 0 70 8 26.25-20.5 46.75-30.25T376 64q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34 99.5z"/></svg>
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" data-md-component="overlay" for="__drawer"></label>
    
      <a href="#overview" tabindex="0" class="md-skip">
        Skip to content
      </a>
    
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href="../../.." title="St. Jude Cloud Documentation" aria-label="St. Jude Cloud Documentation" class="md-header-nav__button md-logo">
          
            <img alt="logo" src="../../../images/stjude-logo-child-white.png" width="40" height="28">
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="__drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <div class="md-flex__ellipsis md-header-nav__title" data-md-component="title">
          
            <span class="md-header-nav__topic">
              St. Jude Cloud Documentation
            </span>
            <span class="md-header-nav__topic">
              
                NeoepitopePred
              
            </span>
          
        </div>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <label class="md-icon md-icon--search md-header-nav__button" for="__search"></label>
          
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" aria-label="search" name="query" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query" data-md-state="active">
      <label class="md-icon md-search__icon" for="__search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">
        &#xE5CD;
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
        
      </div>
      
        <div class="md-flex__cell md-flex__cell--shrink">
          <div class="md-header-nav__source">
            


  

<a href="https://github.com/stjudecloud/docs/" title="Go to repository" class="md-source" data-md-source="github">
  
    <div class="md-source__icon">
      <svg viewBox="0 0 24 24" width="24" height="24">
        <use xlink:href="#__github" width="24" height="24"></use>
      </svg>
    </div>
  
  <div class="md-source__repository">
    stjudecloud/docs
  </div>
</a>
          </div>
        </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
        
      
      
      <main class="md-main" role="main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="__drawer">
    <a href="../../.." title="St. Jude Cloud Documentation" class="md-nav__button md-logo">
      
        <img alt="logo" src="../../../images/stjude-logo-child-white.png" width="48" height="48">
      
    </a>
    St. Jude Cloud Documentation
  </label>
  
    <div class="md-nav__source">
      


  

<a href="https://github.com/stjudecloud/docs/" title="Go to repository" class="md-source" data-md-source="github">
  
    <div class="md-source__icon">
      <svg viewBox="0 0 24 24" width="24" height="24">
        <use xlink:href="#__github" width="24" height="24"></use>
      </svg>
    </div>
  
  <div class="md-source__repository">
    stjudecloud/docs
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="../../.." title="Welcome to St. Jude Cloud" class="md-nav__link">
      Welcome to St. Jude Cloud
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-2" type="checkbox" id="nav-2">
    
    <label class="md-nav__link" for="nav-2">
      COVID-19 Quickstart
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-2">
        COVID-19 Quickstart
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../covid-19/" title="Introduction" class="md-nav__link">
      Introduction
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../covid-19/getting-started/" title="Getting started" class="md-nav__link">
      Getting started
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../covid-19/upload-local/" title="Data upload (local)" class="md-nav__link">
      Data upload (local)
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../covid-19/upload-cluster/" title="Data upload (HPC)" class="md-nav__link">
      Data upload (HPC)
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../covid-19/creating-a-cloud-app-covid/" title="Creating a cloud app" class="md-nav__link">
      Creating a cloud app
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../covid-19/interactive-node/" title="Interactive nodes" class="md-nav__link">
      Interactive nodes
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../covid-19/visualizing-ngs-data/" title="Visualizing NGS data" class="md-nav__link">
      Visualizing NGS data
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../account-and-billing/" title="Account and Billing" class="md-nav__link">
      Account and Billing
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-4" type="checkbox" id="nav-4">
    
    <label class="md-nav__link" for="nav-4">
      Data Guides
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-4">
        Data Guides
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../data/about-our-data/" title="About Our Data" class="md-nav__link">
      About Our Data
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../data/data-request/" title="Making a Data Request" class="md-nav__link">
      Making a Data Request
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../data/working-with-our-data/" title="Working with Our Data" class="md-nav__link">
      Working with Our Data
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../data/creating-a-cloud-app/" title="Creating a Cloud App" class="md-nav__link">
      Creating a Cloud App
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../data/data-transfer-app/" title="Data Transfer Application" class="md-nav__link">
      Data Transfer Application
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../data/command-line/" title="Command Line Interaction" class="md-nav__link">
      Command Line Interaction
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-5" type="checkbox" id="nav-5" checked>
    
    <label class="md-nav__link" for="nav-5">
      Tool Guides
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-5">
        Tool Guides
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../pecan-pie/" title="Pecan PIE" class="md-nav__link">
      Pecan PIE
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="__toc">
    
    
      <label class="md-nav__link md-nav__link--active" for="__toc">
        NeoepitopePred
      </label>
    
    <a href="./" title="NeoepitopePred" class="md-nav__link md-nav__link--active">
      NeoepitopePred
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#overview" class="md-nav__link">
    Overview
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#getting-started" class="md-nav__link">
    Getting started
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#input-file-configuration" class="md-nav__link">
    Input file configuration
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#uploading-data" class="md-nav__link">
    Uploading data
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#running-the-tool" class="md-nav__link">
    Running the tool
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#selecting-parameters" class="md-nav__link">
    Selecting parameters
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hooking-up-inputs" class="md-nav__link">
    Hooking up inputs
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#starting-the-workflow" class="md-nav__link">
    Starting the workflow
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#monitoring-run-progress" class="md-nav__link">
    Monitoring run progress
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#analysis-of-results" class="md-nav__link">
    Analysis of results
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#finding-the-raw-results-files" class="md-nav__link">
    Finding the raw results files
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#interpreting-results" class="md-nav__link">
    Interpreting results
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#frequently-asked-questions" class="md-nav__link">
    Frequently asked questions
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
    
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../chipseq/" title="ChIP-Seq Peak Calling" class="md-nav__link">
      ChIP-Seq Peak Calling
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../rapid-rnaseq/" title="Rapid RNA-Seq Fusion Detection" class="md-nav__link">
      Rapid RNA-Seq Fusion Detection
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../warden/" title="WARDEN Differential Expression Analysis" class="md-nav__link">
      WARDEN Differential Expression Analysis
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../mutational-signatures/" title="Mutational Signatures" class="md-nav__link">
      Mutational Signatures
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../cis-x/" title="Cis-X" class="md-nav__link">
      Cis-X
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-6" type="checkbox" id="nav-6">
    
    <label class="md-nav__link" for="nav-6">
      Visualization Guides
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-6">
        Visualization Guides
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../portals/pecan/" title="PeCan Data Portal" class="md-nav__link">
      PeCan Data Portal
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../portals/genome-paint/" title="Genome Paint" class="md-nav__link">
      Genome Paint
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../portals/sickle-cell/" title="Sickle Cell Genomics Portal" class="md-nav__link">
      Sickle Cell Genomics Portal
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-7" type="checkbox" id="nav-7">
    
    <label class="md-nav__link" for="nav-7">
      Forms
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-7">
        Forms
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../forms/how-to-fill-out-DAA/" title="Filling out a Data Access Agreement" class="md-nav__link">
      Filling out a Data Access Agreement
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../forms/how-to-fill-out-Extension/" title="Filling out an Extension Addendum" class="md-nav__link">
      Filling out an Extension Addendum
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../citing-stjude-cloud/" title="Citing St. Jude Cloud" class="md-nav__link">
      Citing St. Jude Cloud
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../faq/" title="Frequently Asked Questions" class="md-nav__link">
      Frequently Asked Questions
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../glossary/" title="Glossary" class="md-nav__link">
      Glossary
    </a>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#overview" class="md-nav__link">
    Overview
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#getting-started" class="md-nav__link">
    Getting started
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#input-file-configuration" class="md-nav__link">
    Input file configuration
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#uploading-data" class="md-nav__link">
    Uploading data
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#running-the-tool" class="md-nav__link">
    Running the tool
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#selecting-parameters" class="md-nav__link">
    Selecting parameters
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hooking-up-inputs" class="md-nav__link">
    Hooking up inputs
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#starting-the-workflow" class="md-nav__link">
    Starting the workflow
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#monitoring-run-progress" class="md-nav__link">
    Monitoring run progress
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#analysis-of-results" class="md-nav__link">
    Analysis of results
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#finding-the-raw-results-files" class="md-nav__link">
    Finding the raw results files
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#interpreting-results" class="md-nav__link">
    Interpreting results
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#frequently-asked-questions" class="md-nav__link">
    Frequently asked questions
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/stjudecloud/docs/edit/master/docs/guides/tools/neoepitope.md" title="Edit this page" class="md-icon md-content__icon">&#xE3C9;</a>
                
                
                  <h1>NeoepitopePred</h1>
                
                <table>
<thead>
<tr>
<th></th>
<th></th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>Authors</strong></td>
<td>Ti-Cheng Chang</td>
</tr>
<tr>
<td><strong>Publication</strong></td>
<td><a href="https://www.ncbi.nlm.nih.gov/pubmed/28854978">The Neoepitope Landscape in Pediatric Cancers. Genome Medicine. 2017. 9.1: 78</a>.</td>
</tr>
<tr>
<td><strong>Technical Support</strong></td>
<td><a href="https://stjude.cloud/contact/">Contact Us</a></td>
</tr>
</tbody>
</table>
<p>Cancers are caused by somatically acquired alterations including single 
nucleotide variations (SNVs), small insertion/deletions (indels),
translocations, and other types of rearrangements. The genes affected by
these mutations may produce altered proteins, some of which may lead to
the emergence of tumor-specific immunogenic epitopes. We developed an
analytical workflow for identification of putative neoepitopes based on
somatic missense mutations and gene fusions using whole genome
sequencing data. The workflow has been used to characterize neoepitope
landscape of 23 subtypes of pediatric cancer in the Pediatric Cancer
Genome Project<sup id="fnref:1"><a class="footnote-ref" href="#fn:1">1</a></sup>.</p>
<h2 id="overview">Overview</h2>
<p><strong>Inputs</strong></p>
<table>
<thead>
<tr>
<th>Name</th>
<th>Type</th>
<th>Description</th>
<th>Example</th>
</tr>
</thead>
<tbody>
<tr>
<td>FastQ files (<em>required</em> if using FastQ inputs)</td>
<td>Input file</td>
<td>Gzipped FastQ files generated by experiment.</td>
<td>Sample_R1.fastq.gz and Sample_R2.fastq.gz</td>
</tr>
<tr>
<td>BAM files (<em>required</em> if using BAM inputs)</td>
<td>Input file</td>
<td>BAM files aligned against HG19/Hg38 (WGS, WES or RNA-Seq).</td>
<td>Sample.bam</td>
</tr>
<tr>
<td>BAM indices (<em>required</em> if using BAM inputs)</td>
<td>Input file</td>
<td>Corresponding BAM index of the BAM files above.</td>
<td>Sample.bam.bai</td>
</tr>
<tr>
<td>Mutation file (<em>required</em>)</td>
<td>Input file</td>
<td>File describing the mutations present in the sample (special format, see below).</td>
<td>*.txt (tab-delimited)</td>
</tr>
<tr>
<td>SNV or fusion</td>
<td>Parameter</td>
<td>Specify the mutation file contains SNV or gene fusion.</td>
<td>SNV</td>
</tr>
<tr>
<td>Peptide size</td>
<td>Parameter</td>
<td>Size of the peptide.</td>
<td>9</td>
</tr>
<tr>
<td>Affinity threshold</td>
<td>Parameter</td>
<td>Affinity cutoff for epitope prediction report.</td>
<td>500</td>
</tr>
</tbody>
</table>
<p><strong>Outputs</strong></p>
<table>
<thead>
<tr>
<th>Name</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>Epitope affinity prediction (html)</td>
<td>Epitope affinity. The peptide with affinity &lt; cutoff will be highlighted.</td>
</tr>
<tr>
<td>Epitope affinity prediction (xlsx)</td>
<td>Excel tables for the information of all epitopes</td>
</tr>
<tr>
<td>Affinity (raw output)</td>
<td>Epitope affinity</td>
</tr>
<tr>
<td>Peptide sequence (raw output)</td>
<td>Peptide sequences in Fasta format</td>
</tr>
</tbody>
</table>
<p><strong>HLA Typing Algorithm</strong></p>
<p>The HLA typing algorithm is used to predict the HLA class I alleles.
Users can either provide FastQ (paired or single end reads) or a BAM
file as input. When using a BAM file as input, the reads surrounding the
HLA loci and unmapped reads will be extracted. The reads will be fed
into Optitype for HLA typing. The default settings for Optitype are
used. The output of the HLA type can be combined with the our epitope
detection algorithm to perform affinity prediction of neoepitopes.</p>
<p>If you use <strong>FastQ</strong> files as input:  </p>
<ol>
<li>The input FastQs will be aligned against the Optitype HLA reference
    sequences using razers3 (see <a href="https://github.com/FRED-2/OptiType">https://github.com/FRED-2/OptiType</a>).</li>
<li>The fished FastQs will be used for HLA typing using Opitype.</li>
</ol>
<p>If you use <strong>BAM</strong> files as input:  </p>
<ol>
<li>The reads falling within the HLA loci and their paralogous loci will
    be extracted.</li>
<li>The reads unmapped to the human genome will be extracted.</li>
<li>The reads from step 1 and 2 will be combined and deduplicated (in
    FastQ format).</li>
<li>The input FastQs will be aligned against the Optitype HLA reference
    sequences using razers3 (see <a href="https://github.com/FRED-2/OptiType">https://github.com/FRED-2/OptiType</a>).</li>
<li>The fished FastQs will be used for HLA typing using Opitype.</li>
</ol>
<p><strong>Epitope Prediction Algorithm</strong></p>
<p>The epitope prediction algorithm first extracts peptides covering an array
of tiling peptides (size defined by users) overlapping each missense
mutation or gene fusion. Fusion junctions can be identified using RNA-Seq
by fusion detection tools (Li et. al, unpublished). NetMHCcons<sup id="fnref:3"><a class="footnote-ref" href="#fn:3">3</a></sup> is subsequently 
used to predict affinities of the peptide array for each HLA receptor in 
each sample. The neoepitope with affinity lower than the threshold will 
be highlighted in output file (default 500 nM). Below is an outline of internal steps the algorithm performs in order to generate the final report.</p>
<ol>
<li>Check the version of the genomic position of the input SNV/fusion
    file.</li>
<li>Lift over the genomic coordinations if the reference genomic
    position is not HG19. Currently, the internal genome annotation was
    based on HG19 and the genome coordinates of the mutation files will
    be adjusted to HG19 for peptide extraction.</li>
<li>Extract the peptide flanking the mutations.</li>
<li>Run NetMHCcons to obtain the affinity prediction of the peptides.</li>
<li>Produce the affinity report of each peptide.</li>
</ol>
<h2 id="getting-started">Getting started</h2>
<p>To get started, you need to navigate to the <a href="https://platform.stjude.cloud/tools/neoepitope">NeoepitopePred tool page</a>. You'll need to click
the "Start" button in the left hand pane. This creates a cloud workspace
in DNAnexus with the same name as the tool. After this, you will be able 
to upload your input files to that workspace.</p>
<p><img alt="" src="../../../images/guides/tools/neoepitope/click-start.gif" /></p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If you can't see the "Start" button, one of these two scenarios is likely the case:</p>
<ul>
<li>You see three buttons on the left sidebar instead of one. In this case,
  you've already clicked the "Start" button previously, and a cloud workspace has
  already been created for you. In this case, you're good! You can move
  on to the next section.</li>
<li>If you cannot see <em>any</em> buttons on the left side, you probably have not
  logged in yet. If you see a sentence that says "Log in to launch this 
  tool", simply login and try again.</li>
</ul>
<p>If neither of these are the case and you still can't click "Start",
<a href="https://stjude.cloud/contact">contact us</a>.</p>
</div>
<h3 id="input-file-configuration">Input file configuration</h3>
<p>Users need to provide a mutation file for SNV or gene fusion. The format
of the mutation file is shown in the following example. The file can be
prepared in Excel and saved as a tab-delimited text file to use as
input.</p>
<p>The HLA alleles for testing will be derived from the HLA typing module
using the workflow. The peptide size and affinity cutoff can be modified
by users. </p>
<p><strong>Mutation file format</strong></p>
<table>
<thead>
<tr>
<th>GeneName</th>
<th>Sample</th>
<th>Chr</th>
<th>Postion_hg19</th>
<th>Class</th>
<th>AAChange</th>
<th>mRNA_acc</th>
<th>ReferenceAllele</th>
<th>MutantAllele</th>
</tr>
</thead>
<tbody>
<tr>
<td>Gene1</td>
<td>SampleA</td>
<td>chr10</td>
<td>106150600</td>
<td>missense</td>
<td>R663H</td>
<td>NM_00101</td>
<td>A</td>
<td>T</td>
</tr>
<tr>
<td>Gene2</td>
<td>SampleA</td>
<td>chr2</td>
<td>32330151</td>
<td>missense</td>
<td>N329N</td>
<td>NM_00102</td>
<td>T</td>
<td>G</td>
</tr>
</tbody>
</table>
<div class="admonition example">
<p class="admonition-title">Notes on preparing the above file</p>
<ul>
<li>The chromosome requires a 'chr' prefix.</li>
<li>The position requires a suffix of HG19/HG38 to indicate the human genome assembly version.</li>
<li>Only the missense mutations/gene fusion is supported currently and
    the other types of mutations will not be processed.</li>
</ul>
</div>
<p><strong>Mutation file example</strong></p>
<p><img alt="" src="../../../images/guides/tools/neoepitope/mutation-file-example.png" /></p>
<h2 id="uploading-data">Uploading data</h2>
<p>NeoepitopePred takes the following files as input:</p>
<ul>
<li>A pair of Gzipped FastQ files or an HG19/HG38 aligned BAM file. These can be
  generated from whole genome sequencing, whole exome sequencing, or RNA-Seq. </li>
<li>A file describing the mutations in a sample. </li>
</ul>
<p>You can upload these files using the <a href="../../data/data-transfer-app/">data transfer application</a>
or by uploading them through <a href="../../data/command-line/">the command line</a>.
Both of the guides linked here will contain more details on how to upload
data using that method, so we defer to those guides here.</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>If you plan to upload data through the St. Jude Cloud Data Transfer application
(recommended), you can click the "Upload Data" button in the left panel. If you
have not already downloaded the app, do so by clicking "Download app". Once you
have the app, you can click "Open app" to open the app with the tool's cloud 
workspace already opened and ready to drag-and-drop files into it!</p>
<p>For more information, check out the <a href="../../data/data-transfer-app/">data transfer application</a> guide.</p>
</div>
<h2 id="running-the-tool">Running the tool</h2>
<div class="admonition caution">
<p class="admonition-title">Caution</p>
<p>This pipeline assumes HG19 coordinates in the mutation file. If the
coordinates are based on HG38, the coordinates will lifted over to HG19
to perform epitope affinity prediction.</p>
</div>
<p>Once you've uploaded data to your cloud workspace, 
click "Launch Tool" on the <a href="https://platform.stjude.cloud/tools/neoepitope">tool's landing page</a>. 
A dropdown will present the different presets for running the workflow. Here, you can select whether you wish to start with FastQ files or a BAM file.</p>
<p><img alt="" src="../../../images/guides/tools/neoepitope/launch-tool.gif" /></p>
<h3 id="selecting-parameters">Selecting parameters</h3>
<p>There are a number of other parameters that can be customized. To 
see the options available, click the gear cog next to the 
"Neoepitope Prediction" substep. For a full list of the parameters and their
descriptions, see <a href="#input">the input section</a> (specifically, you are 
looking at the items in the table labeled "parameters").</p>
<p><img alt="" src="../../../images/guides/tools/neoepitope/change-parameters.gif" /></p>
<h3 id="hooking-up-inputs">Hooking up inputs</h3>
<p>Next, you'll need to hook up the FastQ/BAM and mutation files you uploaded in 
<a href="#uploading-data">the upload data section</a>. You can do this by 
clicking on the <code class="codehilite"><span class="err">BAM alignment file</span></code> and <code class="codehilite"><span class="err">BAM index file</span></code> and <code class="codehilite"><span class="err">Mutation array</span></code> slots and
selecting the respective files.</p>
<p><img alt="" src="../../../images/guides/tools/neoepitope/hookup-inputs.gif" /></p>
<h3 id="starting-the-workflow">Starting the workflow</h3>
<p>Once your input files are hooked up, you should be able to start the workflow
by clicking the "Run as Analysis..." button in the top right hand corner
of the workflow dialog.</p>
<p><img alt="" src="../../../images/guides/tools/neoepitope/run-analysis.gif" /></p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>If you cannot click this button, please ensure that all of the inputs are correctly hooked up (see <a href="#hooking-up-inputs">hooking up inputs</a>).</p>
<p>If you're still have trouble, please <a href="https://stjude.cloud/contact">contact us</a> and include
a screenshot of the workflow screen above.</p>
</div>
<h2 id="monitoring-run-progress">Monitoring run progress</h2>
<p>Once you have started one or more NeoepitopePred runs, you can safely close your
browser and come back later to check the status of the jobs. To do this,
navigate to the <a href="https://platform.stjude.cloud/tools/neoepitope">tool's landing page</a>. 
Next, click "View Results" then select the "View Running Jobs" option. 
You will be redirected to the job monitoring page. Each job you kicked off
gets one row in this table.</p>
<p><img alt="" src="../../../images/guides/tools/neoepitope/monitoring-jobs.gif" /> </p>
<p>You can click the "+" on any of the runs to check 
the status of individual steps of the ChIP-Seq pipeline.
Other information, such as time, cost of individual steps in the pipeline, 
and even viewing the job logs can accessed by clicking around the sub-items.</p>
<p><img alt="" src="../../../images/guides/tools/neoepitope/job-detailed-view.gif" /> </p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>Power users can refer to the <a href="https://documentation.dnanexus.com/user/running-apps-and-workflows/monitoring-executions">DNAnexus Monitoring Executions Documentation</a> for advanced capabilities for monitoring jobs.</p>
</div>
<h2 id="analysis-of-results">Analysis of results</h2>
<p>Each tool in St. Jude Cloud produces a visualization that makes understanding
results more accessible than working with excel spreadsheet or tab delimited
files. This is the primary way we recommend you work with your results. We also
include the raw output files for you to dig into if the visualization is not 
sufficient to answer your research question.</p>
<h3 id="finding-the-raw-results-files">Finding the raw results files</h3>
<p>Navigate to the <a href="https://platform.stjude.cloud/tools/neoepitopepred">tool's landing page</a>. 
In the left hand pane, click "View Results" then "View Results Files". You will
be taken to the filesystem view your cloud workspace. This is similar to your the
filesystem on your computer, and you can do many common operations such as deleting,
renaming, and moving files.</p>
<p><img alt="" src="../../../images/guides/tools/neoepitope/neo-pred-results.png" /> </p>
<h3 id="interpreting-results">Interpreting results</h3>
<p><strong>HLA typing</strong></p>
<p>The output of this app contain the prediction of the HLA class I alleles
from OptiType.</p>
<ul>
<li>A folder stamped with the time will present in the output folder
  (optitype), which contains the raw output.</li>
</ul>
<p><img alt="" src="../../../images/guides/tools/neoepitope/hlatype-result.png" /></p>
<ul>
<li>The file contains the predicted HLA alleles of the sample.</li>
</ul>
<p><img alt="" src="../../../images/guides/tools/neoepitope/hlatype-result2.png" /></p>
<p><strong>Neoepitope prediction</strong></p>
<p>The output contains one summary HTML, one folder with raw output, and one
folder with outputs in Excel formats:</p>
<p><img alt="" src="../../../images/guides/tools/neoepitope/epitope-result.png" /></p>
<p><strong>Epitope_affinity_prediction.html</strong> (shown below):   </p>
<ul>
<li>This file provides a summary of the epitope prediction that can be visualized directly from web browser.  </li>
<li>The peptides with affinity lower than user-defined cutoff will be highlighted in green in the webpage.  </li>
</ul>
<p><img alt="" src="../../../images/guides/tools/neoepitope/epitope-result2.png" /></p>
<p><strong>Raw_output</strong> (shown below): this folder contains the raw output of the affinity
prediction. There will two major types files present here: affinity.out and flanking.seq.</p>
<p><img alt="" src="../../../images/guides/tools/neoepitope/epitope-result3.png" /></p>
<p><img alt="" src="../../../images/guides/tools/neoepitope/epitope-result4.png" /></p>
<p><strong>affinity.out</strong>: these files are the prediction results from the
    netMHCcons for each peptide.</p>
<p><img alt="" src="../../../images/guides/tools/neoepitope/epitope-result5.png" /></p>
<p>The following columns will be shown in the output:</p>
<table>
<thead>
<tr>
<th>Column</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>Gene name</td>
<td>the name of the genes</td>
</tr>
<tr>
<td>Sample</td>
<td>the name of the samples</td>
</tr>
<tr>
<td>Chromosome (chr)</td>
<td>the chromosome location of the variation</td>
</tr>
<tr>
<td>Position</td>
<td>the chromosomal position of the variation. Currently, the position will be lifted over to HG19 to ensure correct translation of peptid sequences based on the internal annotation database of the pipeline. Therefore, the position will be labeled as HG19.</td>
</tr>
<tr>
<td>Class</td>
<td>class of the variation</td>
</tr>
<tr>
<td>Reference allele</td>
<td>reference allele at the position</td>
</tr>
<tr>
<td>Mutant allele</td>
<td>mutated allele at the position</td>
</tr>
<tr>
<td>mRNA_acc</td>
<td>NCBI accession number of the mRNA</td>
</tr>
<tr>
<td>Allele</td>
<td>HLA allele tested</td>
</tr>
<tr>
<td>Peptide</td>
<td>the neoepitope sequences tested</td>
</tr>
<tr>
<td>Gene_variant</td>
<td>the gene and variant residues</td>
</tr>
<tr>
<td>1-log50k</td>
<td>Prediction score from netMHCcons</td>
</tr>
<tr>
<td>nM</td>
<td>Affinity as IC50 values in nM</td>
</tr>
<tr>
<td>%Rank</td>
<td>% Rank of prediction score to a set of 200.000 random natural 9mer peptides</td>
</tr>
<tr>
<td>HLAtype</td>
<td>All of the hla alleles predicted in the specific sample</td>
</tr>
</tbody>
</table>
<p><strong>flanking.seq</strong>: these files contain the sequences used for the prediction.</p>
<p><img alt="" src="../../../images/guides/tools/neoepitope/epitope-result6.png" /></p>
<p><strong>XLSX</strong>: this folder contains the raw output of the affinity prediction as
described above in Excel files. The files can be downloaded and opened with Excel for downstream filtering and analyses.</p>
<p><img alt="" src="../../../images/guides/tools/neoepitope/epitope-result7.png" /></p>
<h2 id="frequently-asked-questions">Frequently asked questions</h2>
<p>None yet! If you have any questions not covered here, feel free to reach
out on <a href="https://hospital.stjude.org/apps/forms/fb/st-jude-cloud-contact/">our contact form</a>.</p>
<div class="footnote">
<hr />
<ol>
<li id="fn:1">
<p>Downing JR, Wilson RK, Zhang J, et al. The Pediatric Cancer Genome
Project. Nature genetics. 2012;44(6):619-622.&#160;<a class="footnote-backref" href="#fnref:1" title="Jump back to footnote 1 in the text">&#8617;</a></p>
</li>
<li id="fn:2">
<p>Szolek A, Schubert B, Mohr C, Sturm M, Feldhahn M, Kohlbacher O:
OptiType: precision HLA typing from next-generation sequencing
data. Bioinformatics 2014, 30:3310-3316.&#160;<a class="footnote-backref" href="#fnref:2" title="Jump back to footnote 2 in the text">&#8617;</a></p>
</li>
<li id="fn:3">
<p>Karosiene E, Lundegaard C, Lund O, Nielsen M: NetMHCcons: a
consensus method for the major histocompatibility complex class I
predictions. Immunogenetics 2012, 64:177-186.&#160;<a class="footnote-backref" href="#fnref:3" title="Jump back to footnote 3 in the text">&#8617;</a></p>
</li>
</ol>
</div>
                
                  
                
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../pecan-pie/" title="Pecan PIE" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                Pecan PIE
              </span>
            </div>
          </a>
        
        
          <a href="../chipseq/" title="ChIP-Seq Peak Calling" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                ChIP-Seq Peak Calling
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            Copyright &copy; 2017 - 2019 St. Jude Children's Research Hospital
          </div>
        
        
        built from <a href="https://github.com/stjudecloud/docs/tree/0119618">0119618</a> using
        
        <a href="https://www.mkdocs.org">MkDocs</a>
        and a modified version of
        <a href="https://squidfunk.github.io/mkdocs-material/">
          Material for MkDocs
        </a>.
      </div>
      
  <div class="md-footer-social">
    <link rel="stylesheet" href="../../../assets/fonts/font-awesome.css">
    
      <a href="https://www.facebook.com/stjuderesearch/" target="_blank" rel="noopener" title="facebook" class="md-footer-social__link fa fa-facebook"></a>
    
      <a href="https://twitter.com/stjuderesearch?lang=en" target="_blank" rel="noopener" title="twitter" class="md-footer-social__link fa fa-twitter"></a>
    
      <a href="https://www.linkedin.com/company/st-jude-childrens-research-hospital" target="_blank" rel="noopener" title="linkedin" class="md-footer-social__link fa fa-linkedin"></a>
    
      <a href="https://github.com/stjude" target="_blank" rel="noopener" title="github" class="md-footer-social__link fa fa-github"></a>
    
  </div>

    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../../assets/javascripts/application.e21ee79b.js"></script>
      
      <script>app.initialize({version:"1.1",url:{base:"../../.."}})</script>
      
    
  </body>
</html>